---
title: "Thyroides carcinoma"
author: "Lydia Fortea, Helena Rodriguez, Julia Mir"
date: "25/4/2018"
output: html_document
---
---
output:
  BiocStyle::html_document
---

<!---
The following chunk of code, which should not be shown in the resulting document (echo=FALSE)
sets up global processing options, such as forcing 'knitr' to stop when an error
in the R code is encountered, caching of the results in the 'cache'
directory and asking 'knitr' to figure out automatically the dependencies among
code chunks to re-calculate cached results (autodep=TRUE).

Other options could be changing the name of the directory where figures end up
('figure' by default), etc. For a full account of 'knitr' options please consult
http://yihui.name/knitr/options

At the end of the chunk a 'cat()' call is made to dump a CSS file that gives
a better look-and-feel than the knitr default one. See the source css/ieo.css
and the resulting projectTemplate.html to understand where this is being dumpted.
--->


```{r setup, cache=FALSE, echo=FALSE, results='hide', message=FALSE}
library(knitr)

opts_chunk$set(cache=TRUE,
               cache.path="cache/QA",
               cache.extra=R.version.string,
               autodep=TRUE,
               fig.align="center",
               comment="")
```

# Quality assessment

## Data import

We start importing the raw table of counts.

<!--
The option 'message=FALSE' avoid dumping R messages such as "Loading required package: methods"
into the output of the report.
-->

```{r, message=FALSE}
library(SummarizedExperiment)

thca <- readRDS("seTHCA.rds")
thca
```

Explore the column (phenotypic) data, which in this case corresponds to clinical
variables, and their corresponding metadata.

```{r}
dim(colData(thca))
colData(thca)[1:5, 1:5]
mcols(colData(thca), use.names=TRUE)
metadata(thca)
```

These metadata consists of two columns of information about the clinical variables.
One called `labelDescription` contains a succint description of the variable, often
not more self-explanatory than the variable name itself, and the other called
'CDEID' corresponds to the so-called `Common Data Element (CDE)` identifier. This
identifier can be use in https://cdebrowser.nci.nih.gov to search for further
information about the associated clinical variable using the `Advanced search`
form and the `Public ID` attribute search.

Now, explore the row (feature) data.

```{r}
rowData(thca)
rowRanges(thca)
```

To perform quality assessment and normalization we need first to load the
[edgeR](http://bioconductor.org/packages/edgeR) R/Bioconductor package and
create a `DGEList' object.

```{r, message=FALSE}
library(edgeR)

dge <- DGEList(counts=assays(thca)$counts, genes=mcols(thca))
saveRDS(dge, file.path("./", "dge.rds"))
```

Now calculate $\log_2$ CPM values of expression and put them as an additional
assay element to ease their manipulation.

```{r}
assays(thca)$logCPM <- cpm(dge, log=TRUE, prior.count=0.5)
assays(thca)$logCPM[1:5, 1:5]
```

## Sequencing depth

Let's examine the sequencing depth in terms of total number of sequence read counts
mapped to the genome per sample. Figure \@ref(fig:libsizes) below shows the
sequencing depth per sample, also known as library sizes, in increasing order.

<!---
you can control the height and width in pixels of the figure with 'out.height' and
'out.width'. Figures are automatically numbered, to refer to them in the main test
you should use the notation shown above as \@ref(fig:xxxx) with xxxx being the label
in the code chunk that also gives the filename of the figure. This name must be unique
--->

```{r libsizes, echo=FALSE, fig.cap="Library sizes in increasing order."}
ord <- order(dge$sample$lib.size/1e6)
barplot(dge$sample$lib.size[ord]/1e6, las=1, ylab="Millions of reads",
                xlab="Samples", col=c("red", "blue")[(thca$type[ord] == "tumor") + 1], border = NA)
legend("topleft", c("tumor", "normal"), fill=c("blue", "red"), inset=0.01)
```

Tumor samples: 513
Normal samples: 59

As we can see in this plot, we have too more reads from tumor than normal one with similar depth.
This figure reveals substantial differences in sequencing depth between samples
and we may consider discarding those samples whose depth is substantially lower
than the rest. To identify who are these samples we may simply look at the
actual numbers including portion of the sample identifier that distinguishes them.

**GENDER**
As we have NA values in the gender of several samples of the dataset, so we need first to remove those values in order to differenciate male/female.

##LOOOK

```{r libsizes, echo=FALSE, fig.cap="Library sizes in increasing order."}
ord <- order(dge$sample$lib.size/1e6)
barplot(dge$sample$lib.size[ord]/1e6, las=1, ylab="Millions of reads",
                xlab="Samples", col=c("red", "blue")[(thca$gender[ord])], border = NA)
legend("topleft", c("male", "female"), fill=c("blue", "red"), inset=0.01)
```
We have 140 samples from male, and 406 samples of female.

**Sample Depth**

We want to remove those genes which less than 40 of sequencing depth.

```{r}
sampledepth <- round(dge$sample$lib.size / 1e6, digits=1)
names(sampledepth) <- substr(colnames(thca), 6, 12)
sort(sampledepth)
```

Now we can filter the data set by sample depth and NA values in gender.

```{r}
maskd <- sampledepth > 40 #sample depth filter
maskg <- !is.na(thca$gender) #gender filter
dim(thca)
thca.filt <- thca[maskd,maskg]
dge.filt <- dge[maskd,maskg]
dim(thca.filt)
```

## Distribution of expression levels among samples

Let's look at the distribution of expression values per sample in terms of
logarithmic CPM units. Due to the large number of samples, we display tumor
and normal samples separately, and are shown in Figure \@ref(fig:distRawExp)

<!---
the option echo=FALSE hides the R code. When plotting in general one does not
want to see the code. Options fig.height and fig.width control height and width
of the plot in inches while out.height and out.width do it in the final output
file; see http://yihui.name/knitr/options for full details.
--->

```{r distRawExp, echo=FALSE, fig.height=4, fig.width=10, out.width="800px", fig.cap="Non-parametric density distribution of expression profiles per sample.", message=FALSE}
library(geneplotter)
par(mfrow=c(1, 2))
multidensity(as.list(as.data.frame(assays(thca.filt[, thca.filt$type == "tumor"])$logCPM)),
                          xlab="log 2 CPM", legend=NULL, main="Tumor samples", las=1)
multidensity(as.list(as.data.frame(assays(thca.filt[, thca.filt$type == "normal"])$logCPM)),
                          xlab="log 2 CPM", legend=NULL, main="Normal samples", las=1)
```

*Explanation*

*GENDER*

Now, we also perform the multidensity plots between males and females, to know if there is a difference between them.

```{r distRawExp, echo=FALSE, fig.height=4, fig.width=10, out.width="800px", fig.cap="Non-parametric density distribution of expression profiles per sample.", message=FALSE}
library(geneplotter)
par(mfrow=c(1, 2))
multidensity(as.list(as.data.frame(assays(thca.filt[, thca.filt$gender == "FEMALE"])$logCPM)),
                          xlab="log 2 CPM", legend=NULL, main="Female samples", las=1)
multidensity(as.list(as.data.frame(assays(thca.filt[, thca.filt$gender == "MALE"])$logCPM)),
                          xlab="log 2 CPM", legend=NULL, main="Male samples", las=1)
```

```{r}
boxplot(assays(thca.filt)$logCPM, col = "gray", xlab = "Samples", ylab = expression(log[2] *
"CPM"), cex.axis = 1.2, cex.lab = 1.5, las = 1)
```
*Explanation*

## Distribution of expression levels among genes

Let's calculate now the average expression per gene through all the samples.
Figure \@ref(fig:exprdist) shows the distribution of those values across genes.

```{r exprdist, echo=FALSE, out.width="400px", fig.cap="Distribution of average expression level per gene."}
avgexp <- rowMeans(assays(thca.filt)$logCPM)
hist(avgexp, xlab="log2 CPM", main="", las=1, col = "blue")
abline(v=1, col="red", lwd=2)
```

## Filtering of lowly-expressed genes

In the light of this plot, we may consider a cutoff of 1 log CPM unit as minimum value
of expression to select genes being expressed across samples. Using this cutoff we proceed
to filter out lowly-expressed genes.

```{r}
cpmcutoff <- round(10/min(dge.filt$sample$lib.size/1e+06), digits = 1)
cpmcutoff
nsamplescutoff <- min(table(thca.filt$gender))
nsamplescutoff
mask <- rowSums(cpm(dge.filt) > cpmcutoff) >= nsamplescutoff
thca.filt2 <- thca.filt[mask, ]
dge.filt2 <- dge.filt[mask, ]
dim(thca.filt2)
```

```{r}
par(mar = c(4, 5, 1, 1))
h <- hist(avgexp, xlab = expression("Expression level (" * log[2] * "CPM)"), main = "",
las = 1, col = "grey", cex.axis = 1.2, cex.lab = 1.5)
x <- cut(rowMeans(assays(thca.filt2)$logCPM), breaks = h$breaks)
lines(h$mids, table(x), type = "h", lwd = 10, lend = 1, col = "darkred")
legend("topright", c("All genes", "Filtered genes"), fill = c("grey", "darkred"))
```


```{r}
mask <- avgexp > 4
dim(thca.filt2)
thca.filt3 <- thca.filt[mask, ]
dim(thca.filt3)
dge.filt3 <- dge.filt[mask, ]
dim(dge.filt3)
```

Visualize the filtered genes

```{r}
par(mar = c(4, 5, 1, 1))
h <- hist(avgexp, xlab = expression("Expression level (" * log[2] * "CPM)"), main = "",
las = 1, col = "grey", cex.axis = 1.2, cex.lab = 1.5)
x <- cut(rowMeans(assays(thca.filt3)$logCPM), breaks = h$breaks)
lines(h$mids, table(x), type = "h", lwd = 10, lend = 1, col = "darkred")
legend("topright", c("All genes", "Filtered genes"), fill = c("grey", "darkred"))
```
We have performed both types of filtering, we get the second filtering as it is more strict.

Store un-normalized versions of the filtered expression data.

```{r}
saveRDS(thca.filt3, file.path("./", "thca.filt.unnorm.rds"))
saveRDS(dge.filt3, file.path("./", "dge.filt.unnorm.rds"))
```

## Normalization

We calculate now the normalization factors on the filtered expression data set.

```{r}
dge.filt3 <- calcNormFactors(dge.filt3)
head(dge.filt3$samples$norm.factors)
```

Replace the raw log2 CPM units in the corresponding assay element of the `SummarizedExperiment`
object, by the normalized ones.

```{r}
assays(thca.filt3)$logCPM <- cpm(dge.filt3, log=TRUE, normalized.lib.sizes=TRUE, prior.count=0.25)
```

Store normalized versions of the filtered expression data.

```{r}
saveRDS(thca.filt3, file.path("./", "thca.filt.rds"))
saveRDS(dge.filt3, file.path("./", "dge.filt.rds"))
```

## MA-plots

We examine now the MA-plots of the normalized expression profiles. We look first to
the tumor samples in Figure \@ref(fig:maPlotsTumor).

<!---
Here we make a MA-plot for each sample. The options 'fig.height' and 'fig.width'
control the relative image size in *inches*. The final image size results from
'height'x'dpi' and 'width'x'dpi', where 'dpi' is the image resolution in
"dots per inch" (by default dpi=72). To scale the image to a desired size use
'out.width' and 'out.height'. More information at http://yihui.name/knitr/options
--->

We first perform a global MA plot, to have a general idea of the data filtered and non-filtered
```{r}
dge$samples$group <- thca$type
table(dge$samples$gorup)
dge.filt3$samples$group <- thca.filt3$type
plotSmear(dge, lowess = TRUE, las = 1, cex.lab = 1.5, cex.axis = 1.2)
abline(h = 0, col = "blue", lwd = 2)
plotSmear(dge.filt3, lowess = TRUE, las = 1, cex.lab = 1.5, cex.axis = 1.2)
abline(h = 0, col = "blue", lwd = 2)
```

We visualize MA plot for each gene in tumor/normal types
```{r maPlotsTumor, fig.height=36, fig.width=6, dpi=100, echo=FALSE, fig.cap="MA-plots of the tumor samples."}
par(mfrow=c(22, 3), mar=c(4, 5, 3, 1))
setmp <- thca.filt3[, thca.filt3$type == "tumor"]
dgetmp <- dge.filt3[, thca.filt3$type == "tumor"]
for (i in 1:ncol(setmp)) {
  A <- rowMeans(assays(setmp)$logCPM)
  M <- assays(setmp)$logCPM[, i] - A
  samplename <- substr(as.character(setmp$bcr_patient_barcode[i]), 1, 12)
  smoothScatter(A, M, main=samplename, las=1)
  abline(h=0, col="blue", lwd=2)
  lo <- lowess(M ~ A)
  lines(lo$x, lo$y, col="red", lwd=2)
}
```

We do not observe samples with major expression-level dependent biases. Let's
look now to the normal samples in Figure \@ref(fig:maPlotsNormal).

```{r maPlotsNormal, fig.height=18, fig.width=6, dpi=100, echo=FALSE, fig.cap="MA-plots of the normal samples."}
par(mfrow=c(9, 3), mar=c(4, 5, 3, 1))
setmp <- thca.filt3[, thca.filt3$type == "normal"]
dgetmp <- dge.filt3[, thca.filt3$type == "normal"]
for (i in 1:ncol(setmp)) {
  A <- rowMeans(assays(setmp)$logCPM)
  M <- assays(setmp)$logCPM[, i] - A
  samplename <- substr(as.character(setmp$bcr_patient_barcode[i]), 1, 12)
  smoothScatter(A, M, main=samplename, las=1)
  abline(h=0, col="blue", lwd=2)
  lo <- lowess(M ~ A)
  lines(lo$x, lo$y, col="red", lwd=2)
}
```

We make a filtering between samples

```{r}
plotMDS(dge.filt3, col = c("red", "blue")[as.integer(dge.filt3$samples$group)], cex = 0.7)
legend("topleft", c("normal", "tumor"), fill = c("red", "blue"), inset = 0.05, cex = 0.7)
```
Tumors near red points, they don't have the same mutations, and they also are not so aggresive.

*Normalization between Samples* LOOK
```{r}
m <- matrix(assays(thca.filt3)$counts, nrow=6654, ncol=555, byrow=TRUE)
dimnames(m) <- list(rownames(thca.filt3), colnames(thca.filt3))
#floor(colMeans(m))
#apply(m, 2, sd)
originalOrder <- apply(m, 2, order)
m <- apply(m, 2, sort)
m <- matrix(rep(floor(rowMeans(m))))
m <- sapply(1:ncol(m), function(x) m[order(originalOrder[, x]), x])
floor(colMeans(m))
apply(m, 2, sd)
```

## Batch identification

We will search now for potential surrogate of batch effect indicators. Given that each sample
names corresponds to a TCGA barcode (see https://wiki.nci.nih.gov/display/TCGA/TCGA+barcode),
following the strategy described in http://bioinformatics.mdanderson.org/main/TCGABatchEffects:Overview
we are going to derive different elements of the TCGA barcode and examine their distribution
across samples.

```{r}
tss <- substr(colnames(thca.filt), 6, 7)
table(tss)
center <- substr(colnames(thca.filt), 27, 28)
table(center)
plate <- substr(colnames(thca.filt), 22, 25)
table(plate)
portionanalyte <- substr(colnames(thca.filt), 18, 20)
table(portionanalyte)
samplevial <- substr(colnames(thca.filt), 14, 16)
table(samplevial)
```

From this information we can make the following observations:

  * All samples were sequenced at the same center (07)

  * In this dataset, we can find six different combinations of tissue type and vial, it can be a potential batch.

  * Samples were collected across different tissue source sites (TSS).

  * Different samples were sequenced with different plates, so it can be also a potential batch.

  * Most of the samples were sequenced using one portion and analyte combinations, but there is other 5 different portion and analyte combinations

From this data we can consider as possible surrogates of batch effect indicator the tssm, plate, portionanalytic and samplevial.

*TSS*
We are going to use the TSS as surrogate of batch effect indicator. Considering our outcome
of interest as molecular changes between sample types, tumor vs. normal, we will examine now
the cross-classification of this outcome with TSS.

```{r}
table(data.frame(TYPE=thca.filt3$type, TSS=tss))
```

Observe that normal tissues from most of TSS are under-represented with respect to the tumor tissues. If TSS is a source of expression variability, this under-representation in the normal samples may lead to a potential confounding effect.

We examine now how samples group together by hierarchical clustering and multidimensional
scaling, annotating the outcome of interest and the the surrogate of batch indicator. We
calculate again log CPM values with a higher prior count to moderate extreme fold-changes
produced by low counts. The resulting dendrogram is shown in Figure \@ref(fig:sampleClustering).

```{r sampleClustering, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S6: Hierarchical clustering of the samples."}
logCPM <- cpm(dge.filt3, log=TRUE, prior.count=3)
d <- as.dist(1-cor(logCPM, method="spearman"))
sampleClustering <- hclust(d)
batch <- as.integer(factor(tss))
sampleDendrogram <- as.dendrogram(sampleClustering, hang=0.1)
names(batch) <- colnames(thca.filt3)
outcome <- paste(substr(colnames(thca.filt3), 9, 12), as.character(thca.filt3$type), sep="-")
names(outcome) <- colnames(thca.filt3)
sampleDendrogram <- dendrapply(sampleDendrogram,
                               function(x, batch, labels) {
                                 if (is.leaf(x)) {
                                   attr(x, "nodePar") <- list(lab.col=as.vector(batch[attr(x, "label")]))
                                   attr(x, "label") <- as.vector(labels[attr(x, "label")])
                                 }
                                 x
                               }, batch, outcome)
plot(sampleDendrogram, main="Hierarchical clustering of samples")
legend("topright", paste("Batch", sort(unique(batch)), levels(factor(tss))), fill=sort(unique(batch)))
```

We can observe there is not evident clustering depending of the tss, so we can descard it as a 
possible batch.

In Figure \@ref(fig:mdsPlot) we show the corresponding MDS plot. Here we see more clearly that the
first source of variation separates tumor from normal samples. We can also observe that two tumor
samples, corresponding to individuals `KL-8404` and `KN-8427` are separated from the rest, just as
it happens in the hierchical clustering. A closer examination of their corresponding MA-plots also
reveals a slight dependence of expression changes on average expression. We may consider discarding
these two samples and doing the MDS plot again to have a closer look to the differences among the rest
of the samples and their relationship with TSS.

```{r mdsPlot, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S7: Multidimensional scaling plot of the samples."}
plotMDS(dge.filt, labels=outcome, col=batch)
legend("bottomleft", paste("Batch", sort(unique(batch)), levels(factor(tss))),
       fill=sort(unique(batch)), inset=0.05)
```

*Plate*
```{r}
table(data.frame(TYPE=thca.filt3$type, plate=plate))
```

```{r sampleClustering, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S8: Hierarchical clustering of the samples by Plate."}
logCPM <- cpm(dge.filt3, log=TRUE, prior.count=3)
d <- as.dist(1-cor(logCPM, method="spearman"))
sampleClustering <- hclust(d)
batch <- as.integer(factor(plate))
sampleDendrogram <- as.dendrogram(sampleClustering, hang=0.1)
names(batch) <- colnames(thca.filt3)
outcome <- paste(substr(colnames(thca.filt3), 9, 12), as.character(thca.filt3$type), sep="-")
names(outcome) <- colnames(thca.filt3)
sampleDendrogram <- dendrapply(sampleDendrogram,
                               function(x, batch, labels) {
                                 if (is.leaf(x)) {
                                   attr(x, "nodePar") <- list(lab.col=as.vector(batch[attr(x, "label")]))
                                   attr(x, "label") <- as.vector(labels[attr(x, "label")])
                                 }
                                 x
                               }, batch, outcome)
plot(sampleDendrogram, main="Hierarchical clustering of samples")
legend("topright", paste("Batch", sort(unique(batch)), levels(factor(plate))), fill=sort(unique(batch)))
```

```{r mdsPlot, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S9: Multidimensional scaling plot of the samples."}
plotMDS(dge.filt, labels=outcome, col=batch)
legend("bottomleft", paste("Batch", sort(unique(batch)), levels(factor(plate))),
       fill=sort(unique(batch)), inset=0.05)
```

*PorrtionAnalyte*

```{r}
table(data.frame(TYPE=thca.filt$type, PA=portionanalyte))
```

```{r sampleClustering, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S10: Hierarchical clustering of the samples by Portion Analyuc."}
logCPM <- cpm(dge.filt3, log=TRUE, prior.count=3)
d <- as.dist(1-cor(logCPM, method="spearman"))
sampleClustering <- hclust(d)
batch <- as.integer(factor(portionanalyte))
sampleDendrogram <- as.dendrogram(sampleClustering, hang=0.1)
names(batch) <- colnames(thca.filt3)
outcome <- paste(substr(colnames(thca.filt3), 9, 12), as.character(thca.filt3$type), sep="-")
names(outcome) <- colnames(thca.filt3)
sampleDendrogram <- dendrapply(sampleDendrogram,
                               function(x, batch, labels) {
                                 if (is.leaf(x)) {
                                   attr(x, "nodePar") <- list(lab.col=as.vector(batch[attr(x, "label")]))
                                   attr(x, "label") <- as.vector(labels[attr(x, "label")])
                                 }
                                 x
                               }, batch, outcome)
plot(sampleDendrogram, main="Hierarchical clustering of samples")
legend("topright", paste("Batch", sort(unique(batch)), levels(factor(portionanalyte))), fill=sort(unique(batch)))
```

```{r mdsPlot, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S11: Multidimensional scaling plot of the samples."}
plotMDS(dge.filt, labels=outcome, col=batch)
legend("bottomleft", paste("Batch", sort(unique(batch)), levels(factor(portionanalyte))),
       fill=sort(unique(batch)), inset=0.05)
```

*Samplevial*

```{r}
table(data.frame(TYPE=thca.filt$type, SV=samplevial))
```

```{r sampleClustering, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S12: Hierarchical clustering of the samples by Portion Analyuc."}
logCPM <- cpm(dge.filt3, log=TRUE, prior.count=3)
d <- as.dist(1-cor(logCPM, method="spearman"))
sampleClustering <- hclust(d)
batch <- as.integer(factor(samplevial))
sampleDendrogram <- as.dendrogram(sampleClustering, hang=0.1)
names(batch) <- colnames(thca.filt3)
outcome <- paste(substr(colnames(thca.filt3), 9, 12), as.character(thca.filt3$type), sep="-")
names(outcome) <- colnames(thca.filt3)
sampleDendrogram <- dendrapply(sampleDendrogram,
                               function(x, batch, labels) {
                                 if (is.leaf(x)) {
                                   attr(x, "nodePar") <- list(lab.col=as.vector(batch[attr(x, "label")]))
                                   attr(x, "label") <- as.vector(labels[attr(x, "label")])
                                 }
                                 x
                               }, batch, outcome)
plot(sampleDendrogram, main="Hierarchical clustering of samples")
legend("topright", paste("Batch", sort(unique(batch)), levels(factor(samplevial))), fill=sort(unique(batch)))
```

From this hierchical clustering plot, we can observe that there is a different cluster depending 
of the samplevial we use, so they can confounding with the interest outcome we want.
```{r mdsPlot, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Figure S11: Multidimensional scaling plot of the samples."}
plotMDS(dge.filt, labels=outcome, col=batch)
legend("bottomleft", paste("Batch", sort(unique(batch)), levels(factor(samplevial))),
       fill=sort(unique(batch)), inset=0.05)
```
REAL BATCH!!!
## Session information

```{r, message=FALSE}
sessionInfo()
```

