---
output:
  BiocStyle::html_document
  bookdown::gitbook: default
  bookdown::pdf_book: default
---


```{r setup, cache=FALSE, echo=FALSE, results='hide', message=FALSE}
library(knitr)

opts_chunk$set(cache=TRUE,
               cache.path="cache/QA",
               cache.extra=R.version.string,
               autodep=TRUE,
               fig.align="center",
               comment="")
```

# Visualising data

The data under study consists in counts of reads for the Thyroid carcinoma as a Summarized Experiment object \@ref(import). This dataset comes from the "Alternative preprocessing of RNA-Sequencing data in The Cancer Genome Atlas leads to improved analysis results" project, with 20115 analyzed genes and 572 samples.

The column data contains pehnotypic data, which corresponds to clinical variables and their corresponding metadata \@ref(coldata), it contains 549 columns with different information about the samples.
The metadata consists of two columns of information about the clinical variables. One called `labelDescription` contains a succint description of the variable, often
not more self-explanatory than the variable name itself, and the other called 'CDEID' corresponds to the so-called `Common Data Element (CDE)` identifier. This
identifier can be use in https://cdebrowser.nci.nih.gov to search for further information about the associated clinical variable using the `Advanced search`
form and the `Public ID` attribute search.

The row data contains feature data \@ref(rowdata). We can observe the characteristics of the studied genes such as their location, ranges, strand, symbol, length and CG content.

```{r import, message=FALSE}
library(SummarizedExperiment)

thca <- readRDS("../seTHCA.rds")
thca
```

```{r coldata}
dim(colData(thca))
colData(thca)[1:5, 1:5]
mcols(colData(thca), use.names=TRUE)
metadata(thca)
```

```{r rowdata}
rowData(thca)
rowRanges(thca)
```

In this supplementary material we will perform a quality assessment and normalization of the data, in order to do that, we will create a 'DGEList' object. 
In order to ease the manipulation of the data $\log_2$ CPM values of expression are calculated \@ref(logCPM).

```{r, message=FALSE}
library(edgeR)

dge <- DGEList(counts=assays(thca)$counts, genes=mcols(thca))
saveRDS(dge, file.path("results", "dge.rds"))
```

```{r logCPM}
assays(thca)$logCPM <- cpm(dge, log=TRUE, prior.count=0.5)
assays(thca)$logCPM[1:5, 1:5]
```

# Quality assessment

## Important Variables

In thyroid cancer we consider some variables more important than others as, according to bibliography, significant differences are seen in the categories of this variables related to thyroid cancer.
We want to have a first look to this categories before the analysis of the whole data. This will give us an idea of the distribution of our data and can be useful in further analysis \@ref(numvars).

```{r numvars, fig.height=20, fig.width=20}
par(mfrow=c(3, 1))
#Type
tumor <- thca[ , colData(thca)$type == "tumor"]
normal <- thca[ , colData(thca)$type == "normal"]
dim(tumor)
dim(normal)
barplot(summary(colData(thca)$type), main="Type of sample", xlab="Sample type", ylab="Total counts")
#Gender
male <- thca[ , !is.na(thca$gender) & colData(thca)$gender == "MALE"]
female <- thca[ , !is.na(thca$gender) & colData(thca)$gender == "FEMALE"]
dim(male)
dim(female)
na <- thca[ , is.na(thca$gender)]
dim(na)
barplot(summary(colData(thca)$gender), main="Patient's gender", xlab="Gender", ylab="Total counts")
#Age
barplot(summary(colData(thca)$age_at_diagnosis),main="Patient's age", xlab="Age", ylab="Total counts")
```
The last "age" column correspond to NA, but we do not remove them in this plot because we may need those samples in further analysis. 
A we can see, we have 513 tumors and 59 normal samples; or, lookig at the gender, 149 males and 406 females, with 17 samples from unknown gender. 
The variable race has not been assessed, although it is considered important, because a big ammount of samples where not defined.

## Sequencing depth

In this part, we examine the sequencing depth in terms of total number of sequence read counts mapped to the genome per sample. Figure \@ref(fig:libsizes) below shows the
sequencing depth per sample, also known as library sizes, in increasing order.

```{r libsizes, echo=FALSE, fig.cap="Library sizes in increasing order.\\label{libsizes}"}
ord <- order(dge$sample$lib.size/1e6)
barplot(dge$sample$lib.size[ord]/1e6, las=1, ylab="Millions of reads", main="Sequencing depth",
                xlab="Samples", col=c("chocolate1", "skyblue")[(thca$type[ord] == "tumor") + 1], border = NA)
legend("topleft", c("tumor", "normal"), fill=c("skyblue", "chocolate1"), inset=0.01)
```

In this plot we can visually observe again that we have more samples from tumor than normal, both types have similar sequencing depth. However, the fugure also reveals substantial differences in sequencing depth between samples, this is why the discardation of those samples whose depth is substantially lower than the rest needs to be taked into consideration.

**Sample Depth**

In order to remove those samples with a lower sequencing depth value, we first calculate the sample depth \@ref(smpdepth), and then select a cutoff to remove samples.

```{r smpdepth}
sampledepth <- round(dge$sample$lib.size / 1e6, digits=1)
names(sampledepth) <- substr(colnames(thca), 6, 12)
sort(sampledepth)
```

Moreover, we know that this cancer has different affection depending on the gender, being women more susceptibles, so maybe we would like to compare samples regarding the gender. Samples with unknown (NA) gender will be removed in order to allow further analysis.

In conclusion, two filters are applied \@ref(maskdg):

- Remove all sample with a sample depth lower than 40.
- Remove samples without known gender.

```{r maskdg}
maskd <- sampledepth > 40 #sample depth filter
maskg <- !is.na(thca$gender) #gender filter
dim(thca)
thca.filt <- thca[,maskd&maskg]
dge.filt <- dge[,maskd&maskg]
dim(thca.filt)
```

Once the filters are applied, the dataset contains 20115 genes and 467 samples. This way, none of our samples has a low sample depth \@ref(fig:filtlibsizes). 

```{r filtlibsizes, echo=FALSE, fig.cap="Library sizes in increasing order.\\label{fig:filtlibsizes}"}
ord <- order(dge.filt$sample$lib.size/1e6)
barplot(dge.filt$sample$lib.size[ord]/1e6, las=1, ylab="Millions of reads", main="Sequencing depth",
                xlab="Samples", col=c("chocolate1", "skyblue")[(thca.filt$type[ord] == "tumor") + 1], border = NA)
legend("topleft", c("tumor", "normal"), fill=c("skyblue", "chocolate1"), inset=0.01)
```

## Distribution of expression levels among samples

One way to normalize RNA-seq data is an adjustment to compare across features 
in a sample, this can be performed using count per million reads (CPM). The distribution of expression values per samle in terms of logarithmic CPM units is ploted \@ref(distRawExp) separating by tumor and normal samples due to the large number of samples.
A box plot of the expression values per samples is also performed \@ref(fig:distboxplot) in order to have another visual way to interpret the data and spot location differences. 

```{r distRawExp, echo=FALSE, fig.height=4, fig.width=10, out.width="800px", fig.cap="Non-parametric density distribution of expression profiles per sample.\\label{fig:distRawExp}", message=FALSE}
library(geneplotter)
par(mfrow=c(1, 2))
multidensity(as.list(as.data.frame(assays(thca.filt[, thca.filt$type == "tumor"])$logCPM)),
                          xlab="log 2 CPM", legend=NULL, main="Tumor samples", las=1)
multidensity(as.list(as.data.frame(assays(thca.filt[, thca.filt$type == "normal"])$logCPM)),
                          xlab="log 2 CPM", legend=NULL, main="Normal samples", las=1)
```

```{r distboxplot, echo=FALSE, fig.height=4, fig.width=12, out.width="800px", fig.cap="Box plot of the distribution expression values across samples."\\label{fig:distboxplot}, message=FALSE}
boxplot(assays(thca.filt)$logCPM, col = "mediumseagreen", xlab = "Samples", ylab = expression(log[2] * "CPM"), cex.axis = 0.5, cex.lab = 1.5, las = 1, axes=F)
```

From this plots, it is observes that all samples follow a similar pattern with a group of highly expressed genes and another group of lowly expressed genes, following the common pattern of expression. From the box plot we can observe that there are no samples that deviate from the average interquartile range.
For all this reasons, we assume that we do not need to normalise among samples.

## Distribution of expression levels among genes

In order to identify lowly expressed genes, the average expression per gene through all samples is calculated. The distribution of thos values across genes is represented \@ref(fig:exprdist).

```{r exprdist, echo=FALSE, out.width="400px", fig.cap="Distribution of average expression level per gene.\\label{fig:exprdist}"}
avgexp <- rowMeans(assays(thca.filt)$logCPM)
hist(avgexp, xlab="log2 CPM", main="", las=1, col = "grey")
abline(v=0, col="red", lwd=2)
```

Most of the genes have at least a log2CPM greater than 0, the ones that are lowly expressed are going to be removed.

## Filtering of lowly-expressed genes

Lowly-expressed genes can be filtered following two criteria:

- Filtering out genes below a minimum average log2CPM throught the samples.
- Filtering out genes with fewer than a given number of sample meeting a minimum log2CPM.

First of all, genes are filtered using the second approach \@ref(fig:cpmcutoff). An histograme is plotted \@ref(fig:expfilt1) in order to visualize the filtering and assess if the filtering is correct and enough.
After the first filtering, the dataset consists of 15408 genes to analyzed. It is seen that many genes have not passed passed the cuttoff (red bars), so they are not filtered out. Also, first red columns are not really representative of the expression level group, so, in order to improve that issue, and taking into account that the number of genes to perform the analysis is enough, the application of a secondary filter is applied.

```{r cpmcutoff}
#calculate cpm cutoff around all samples
cpmcutoff <- round(10/min(dge.filt$sample$lib.size/1e+06), digits = 1)
cpmcutoff
#select number of samples meeting that cutoff
nsamplescutoff <- min(table(thca.filt$type))
nsamplescutoff
mask <- rowSums(cpm(dge.filt) > cpmcutoff) >= nsamplescutoff
thca.filt2 <- thca.filt[mask, ]
dge.filt2 <- dge.filt[mask, ]
dim(thca.filt2)
```

```{r expfilt1, echo=FALSE, out.width="400px", fig.cap="Distribution of average expression level per gene with first filter.\\label{fig:expfilt1}"}
par(mar = c(4, 5, 1, 1))
h <- hist(avgexp, xlab = expression("Expression level (" * log[2] * "CPM)"), 
          main = "Average expression level", ylab ="Frequency", las = 1, col = "grey", 
          cex.axis = 1.2, cex.lab = 1.5)
x <- cut(rowMeans(assays(thca.filt2)$logCPM), breaks = h$breaks)
lines(h$mids, table(x), type = "h", lwd = 10, lend = 1, col = "darkred")
legend("topright", c("All genes", "Filtered genes"), fill = c("grey", "darkred"))
```

In order to perform a further filtering, a cuttoff of 3 log2CPM units is choosed as a minimum value of expression \@ref(cpmcutoff). Another histogram is shoun in order to visualize better the second filtering \@ref(fig:expfilt2).

```{r cpmcutoff2}
mask <- avgexp > 3
dim(thca.filt2)
thca.filt3 <- thca.filt[mask, ]
dim(thca.filt3)
dge.filt3 <- dge.filt[mask, ]
dim(dge.filt3)
```

```{r expfilt2, echo=FALSE, out.width="400px", fig.cap="Distribution of average expression level per gene with second filter.\\label{fig:expfilt2}"}
par(mar = c(4, 5, 1, 1))
h <- hist(avgexp, xlab = expression("Expression level (" * log[2] * "CPM)"), 
          main = "Average expression level", ylab ="Frequency", las = 1, col = "grey", 
          cex.axis = 1.2, cex.lab = 1.5)
x <- cut(rowMeans(assays(thca.filt3)$logCPM), breaks = h$breaks)
lines(h$mids, table(x), type = "h", lwd = 10, lend = 1, col = "darkred")
legend("topright", c("All genes", "Filtered genes"), fill = c("grey", "darkred"))
```

After filtering the dataset with both methods, 9423 genes are left to perform a differential expression analysis.

```{r unnorm}
saveRDS(thca.filt3, file.path("results", "thca.filt.unnorm.rds"))
saveRDS(dge.filt3, file.path("results", "dge.filt.unnorm.rds"))
```

## Normalization

After the filtering for lowly expressed genes, normalization factors need to be calculated in order to allow the detection of differential expression due to biology, not to sequencing depth.

```{r normfact}
dge.filt3 <- calcNormFactors(dge.filt3)
head(dge.filt3$samples$norm.factors)
```

```{r normcpm}
assays(thca.filt3)$logCPM <- cpm(dge.filt3, log=TRUE, normalized.lib.sizes=TRUE, prior.count=0.25)
```

```{r norm}
saveRDS(thca.filt3, file.path("results", "thca.filt.rds"))
saveRDS(dge.filt3, file.path("results", "dge.filt.rds"))
```

## MA-plots

MA plots are used to detect intensity dependent biases, by comparing two groups of the dataset. Firstly, a global MA-plot is examined for formalized and non-normalized data \@ref(fig:maPlotsGeneral), this is done in order to have a general idea of possible dependencies.

```{r maPlotsGeneral, fig.height=6, fig.width=6, dpi=100, echo=FALSE, fig.cap="MA-plots of normalized and non-normalized data.\\label{fig:maPlotsGeneral}"}
dge$samples$group <- thca$type
table(dge$samples$gorup)
dge.filt3$samples$group <- thca.filt3$type
plotSmear(dge, lowess = TRUE, las = 1, cex.lab = 1.5, cex.axis = 1.2, main = "Normalized data", xlab = "Average logCPM", ylab = "logFC: tumor-normal")
abline(h = 0, col = "blue", lwd = 2)
plotSmear(dge.filt3, lowess = TRUE, las = 1, cex.lab = 1.5, cex.axis = 1.2, main = "Non-normalized data", xlab = "Average logCPM", ylab = "logFC: tumor-normal")
abline(h = 0, col = "blue", lwd = 2)
```

In the non-normalized plot, there are lots of genes non differential expreesed (logFC ~ 0) and also several ones with a low log2CPM. These ones are removed in the normalized plot in order to avoid artifacts in posterior analysis. In the normalized plot, a smother shape with less outliers is observed. 

MA-plots of the normalized expression profiles for each tumor \@ref(fig:maPlotsTumor) and normal \@ref(fig:maPlotsNormal) samples sparately are then examined, in order to observe if there is any sample with an anomalous expression profile. 

```{r maPlotsTumor, fig.height=36, fig.width=6, dpi=100, echo=FALSE, fig.cap="MA-plots of the tumor samples.\\label{fig:maPlotsTumor}"}
par(mfrow=c(22, 3), mar=c(4, 5, 3, 1))
setmp <- thca.filt3[, thca.filt3$type == "tumor"]
dgetmp <- dge.filt3[, thca.filt3$type == "tumor"]
for (i in 1:ncol(setmp)) {
  A <- rowMeans(assays(setmp)$logCPM)
  M <- assays(setmp)$logCPM[, i] - A
  samplename <- substr(as.character(setmp$bcr_patient_barcode[i]), 1, 12)
  smoothScatter(A, M, main=samplename, las=1, xlab = "Average logCPM", ylab = "logFC: tumor-normal")
  abline(h=0, col="blue", lwd=2)
  lo <- lowess(M ~ A)
  lines(lo$x, lo$y, col="red", lwd=2)
}
```

```{r maPlotsNormal, fig.height=18, fig.width=6, dpi=100, echo=FALSE, fig.cap="MA-plots of the normal samples.\\label{fig:maPlotsNormal}"}
par(mfrow=c(9, 3), mar=c(4, 5, 3, 1))
setmp <- thca.filt3[, thca.filt3$type == "normal"]
dgetmp <- dge.filt3[, thca.filt3$type == "normal"]
for (i in 1:ncol(setmp)) {
  A <- rowMeans(assays(setmp)$logCPM)
  M <- assays(setmp)$logCPM[, i] - A
  samplename <- substr(as.character(setmp$bcr_patient_barcode[i]), 1, 12)
  smoothScatter(A, M, main=samplename, las=1, xlab = "Average logCPM", ylab = "logFC: tumor-normal")
  abline(h=0, col="blue", lwd=2)
  lo <- lowess(M ~ A)
  lines(lo$x, lo$y, col="red", lwd=2)
}
```

Samples with major expresion-level dependent biasses are not observed in tumor samples. On the other hand, some samples from normal patients presenten an anomalous profile, this samples are removed \@ref(fig:biasses) in order to avoid biasses in the results. In total, 4 patients are removed, as one patitent (TCGA.BJ.A28X) has contributed with both, normal and tumor samples, at the end 5 samples are removed.

```{r biasses}
#select more samples from tumor and normal
maskbad <- substr(colnames(thca.filt3), 0, 12) %in% c("TCGA.BJ.A28X", "TGCA.EL.A3ZP", "TCGA.ET.A3DP","TCGA.KS.A41I")
dim(thca.filt3)
dim(dge.filt3)
thca.filt4 <- thca.filt3[, !maskbad]
dge.filt4 <- dge.filt3[, !maskbad]
dim(dge.filt4)
dim(thca.filt4)
saveRDS(thca.filt4, file.path("results", "thca.filt4.rds"))
saveRDS(dge.filt4, file.path("results", "dge.filt4.rds"))
```

Once normalization is done, a MDS plot \@ref(fig:MDS) (which is a tailored PCA) is performed in order to visualize possible samples with distinctive features from the rest, which can indicate sample potential problems.

```{r MDS, fig.height=6, fig.width=6, dpi=100, echo=FALSE, fig.cap="SMDS of the samples according to its type (tumor or normal).\\label{fig:MDS}"}
plotMDS(dge.filt4, col = c("red", "blue")[as.integer(dge.filt4$samples$group)], cex = 0.7, main ="Sample type MDS", xlab = "Leading logFC dim1", ylab = "Leading logFC dim2")
legend("topleft", c("normal", "tumor"), fill = c("red", "blue"), inset = 0.05, cex = 0.7)
```

Two clusters can be observed. This highlights a difference between normal and tummor samples. A test is needed in order to asses if the cluster occures due to the type of tissue or because there is any batch effect that confounds our outcome of interest. This is done to avoid missundertandments in further analysis.

Moreover, in the cluster created by normal samples some tumor samples are observed. If we discard the bathc effect, leading that the clustering is due to the type of sample, that can be explained becuse not all thyroid tumors have the same mutations, which produces different levels of severity in cancer patients.

## Batch identification

The next step is to search for potential surrogate variables. Given that each sample
name corresponds to a TCGA barcode, different elements are going to be derived of this barcode. Their destribution will be examined across samples. 

```{r}
tss <- substr(colnames(thca.filt4), 6, 7)
table(tss)
center <- substr(colnames(thca.filt4), 27, 28)
table(center)
plate <- substr(colnames(thca.filt4), 22, 25)
table(plate)
portionanalyte <- substr(colnames(thca.filt4), 18, 20)
table(portionanalyte)
samplevial <- substr(colnames(thca.filt4), 14, 16)
table(samplevial)
```

It can be seen that all samples were sequenced at the same center (07). There are six different combinations of tissue type and vial in the dataset, which can be a potential surrogate variable. Samples were collected across different tissue source sites (TSS). Different samples were sequences with different plates, so this can also be a potential confunding variable. Finally, most of the samples were sequenced using one portion and analyte combination, but there are other 5 differentent combinations.

From all this data, it can be considered as possible surrogate variables the tssm, plate, portionalytic and samplevial.

*TSS*
TSS is going to be considered as a possible confunding variable cause of a batch effect. Considering the outcome of interest as molecular changes between sample types, tumor vs. normal,the cross-classification of this outcome with TSS is going to be examined.

```{r tsstable}
table(data.frame(TYPE=thca.filt4$type, TSS=tss))
```

It can be observed that normal tissues from most of TSS are under-represented with respect to the tumor tissues. If TSS is a source of expression variability, this under-representation in the normal samples may lead to a potential confounding effect.

Sample groups are going to be examined together by hierarchical clustering and multidimensional scaling, annotating the outcome of interest and the counfunding indicator. 
The locCPM values are recalculated with a higher prior cound in order to moderate extreme fold-changes produced by low counts.
The resulting dendrogram is shown in Figure \@ref(fig:sampleClustering).

```{r sampleClustering, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Hierarchical clustering of the samples by TSS.\\label{fig:sampleClustering}"}
logCPM <- cpm(dge.filt4, log=TRUE, prior.count=3)
d <- as.dist(1-cor(logCPM, method="spearman"))
sampleClustering <- hclust(d)
batch <- as.integer(factor(tss))
sampleDendrogram <- as.dendrogram(sampleClustering, hang=0.1)
names(batch) <- colnames(thca.filt4)
outcome <- paste(substr(colnames(thca.filt4), 9, 12), as.character(thca.filt4$type), sep="-")
names(outcome) <- colnames(thca.filt4)
sampleDendrogram <- dendrapply(sampleDendrogram,
                               function(x, batch, labels) {
                                 if (is.leaf(x)) {
                                   attr(x, "nodePar") <- list(lab.col=as.vector(batch[attr(x, "label")]))
                                   attr(x, "label") <- as.vector(labels[attr(x, "label")])
                                 }
                                 x
                               }, batch, outcome)
plot(sampleDendrogram, main="Hierarchical clustering of samples: TSS")
legend("topright", paste("Batch", sort(unique(batch)), levels(factor(tss))), fill=sort(unique(batch)))
```

Evident clustering depending of the tss can't be observed, so a batch effect due tu this variable is discarded.

A similar case is observed when analyzing Plate \@ref(fig:sampleClustering2) and PortionAnalyte \@ref(fig:sampleClustering3), a batch effect is not observed as different cathegories do not cluster separately.

```{r platetable}
table(data.frame(TYPE=thca.filt4$type, plate=plate))
```

```{r sampleClustering2, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Hierarchical clustering of the samples by Plate.\\label{fig:sampleClustering2}"}
logCPM <- cpm(dge.filt4, log=TRUE, prior.count=3)
d <- as.dist(1-cor(logCPM, method="spearman"))
sampleClustering <- hclust(d)
batch <- as.integer(factor(plate))
sampleDendrogram <- as.dendrogram(sampleClustering, hang=0.1)
names(batch) <- colnames(thca.filt4)
outcome <- paste(substr(colnames(thca.filt4), 9, 12), as.character(thca.filt4$type), sep="-")
names(outcome) <- colnames(thca.filt4)
sampleDendrogram <- dendrapply(sampleDendrogram,
                               function(x, batch, labels) {
                                 if (is.leaf(x)) {
                                   attr(x, "nodePar") <- list(lab.col=as.vector(batch[attr(x, "label")]))
                                   attr(x, "label") <- as.vector(labels[attr(x, "label")])
                                 }
                                 x
                               }, batch, outcome)
plot(sampleDendrogram, main="Hierarchical clustering of samples: Plate")
legend("topright", paste("Batch", sort(unique(batch)), levels(factor(plate))), fill=sort(unique(batch)))
```

```{r portionanalytetable}
table(data.frame(TYPE=thca.filt4$type, PA=portionanalyte))
```

```{r sampleClustering3, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Hierarchical clustering of the samples by Portion Analytic.\\label{fig:sampleClustering3}"}
logCPM <- cpm(dge.filt4, log=TRUE, prior.count=3)
d <- as.dist(1-cor(logCPM, method="spearman"))
sampleClustering <- hclust(d)
batch <- as.integer(factor(portionanalyte))
sampleDendrogram <- as.dendrogram(sampleClustering, hang=0.1)
names(batch) <- colnames(thca.filt4)
outcome <- paste(substr(colnames(thca.filt4), 9, 12), as.character(thca.filt4$type), sep="-")
names(outcome) <- colnames(thca.filt4)
sampleDendrogram <- dendrapply(sampleDendrogram,
                               function(x, batch, labels) {
                                 if (is.leaf(x)) {
                                   attr(x, "nodePar") <- list(lab.col=as.vector(batch[attr(x, "label")]))
                                   attr(x, "label") <- as.vector(labels[attr(x, "label")])
                                 }
                                 x
                               }, batch, outcome)
plot(sampleDendrogram, main="Hierarchical clustering of samples: Portion Analytic")
legend("topright", paste("Batch", sort(unique(batch)), levels(factor(portionanalyte))), fill=sort(unique(batch)))
```

When analyzin Samplevial \@ref(fig:sampleClustering4), it is observed a clear clusterin of all sampels corresponding to different sample vials.

The corresponding MDS plot \@ref(fig:mdsPlot) is also performed in order to visualize in a different way the clustering of samples, confirming this clustering.

```{r samplevialtable}
table(data.frame(TYPE=thca.filt4$type, SV=samplevial))
```

```{r sampleClustering4, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Hierarchical clustering of the samples by Sample vial.\\label{fig:sampleClustering4}"}
logCPM <- cpm(dge.filt4, log=TRUE, prior.count=3)
d <- as.dist(1-cor(logCPM, method="spearman"))
sampleClustering <- hclust(d)
batch <- as.integer(factor(samplevial))
sampleDendrogram <- as.dendrogram(sampleClustering, hang=0.1)
names(batch) <- colnames(thca.filt4)
outcome <- paste(substr(colnames(thca.filt4), 9, 12), as.character(thca.filt4$type), sep="-")
names(outcome) <- colnames(thca.filt4)
sampleDendrogram <- dendrapply(sampleDendrogram,
                               function(x, batch, labels) {
                                 if (is.leaf(x)) {
                                   attr(x, "nodePar") <- list(lab.col=as.vector(batch[attr(x, "label")]))
                                   attr(x, "label") <- as.vector(labels[attr(x, "label")])
                                 }
                                 x
                               }, batch, outcome)
plot(sampleDendrogram, main="Hierarchical clustering of samples: Sample vial")
legend("topright", paste("Batch", sort(unique(batch)), levels(factor(samplevial))), fill=sort(unique(batch)))
```

Due to that clustering, sample vial is considered to cause a confunding effect with the outcome of interest, it is considered a real batch effect. For that reason, this batch effect is going to be adjusted or removed in order to get a non-biased result.

```{r mdsPlot, fig.height=7, fig.width=14, dpi=100, echo=TRUE, fig.cap="Multidimensional scaling plot of the samples.\\label{fig:mdsPlot}"}
plotMDS(dge.filt4, labels=outcome, col=batch,ain ="Sample vial MDS", xlab = "Leading logFC dim1", ylab = "Leading logFC dim2")
legend("bottomleft", paste("Batch", sort(unique(batch)), levels(factor(samplevial))),
       fill=sort(unique(batch)), inset=0.05)
```

## Session information

```{r, message=FALSE}
sessionInfo()
```
