<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>QAanalysis.utf8</title>
  <meta name="description" content="">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="QAanalysis.utf8" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="QAanalysis.utf8" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="visualising-data.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="visualising-data.html"><a href="visualising-data.html"><i class="fa fa-check"></i><b>1</b> Visualising data</a></li>
<li class="chapter" data-level="2" data-path="quality-assessment.html"><a href="quality-assessment.html"><i class="fa fa-check"></i><b>2</b> Quality assessment</a><ul>
<li class="chapter" data-level="2.1" data-path="quality-assessment.html"><a href="quality-assessment.html#important-variables"><i class="fa fa-check"></i><b>2.1</b> Important Variables</a></li>
<li class="chapter" data-level="2.2" data-path="quality-assessment.html"><a href="quality-assessment.html#sequencing-depth"><i class="fa fa-check"></i><b>2.2</b> Sequencing depth</a></li>
<li class="chapter" data-level="2.3" data-path="quality-assessment.html"><a href="quality-assessment.html#distribution-of-expression-levels-among-samples"><i class="fa fa-check"></i><b>2.3</b> Distribution of expression levels among samples</a></li>
<li class="chapter" data-level="2.4" data-path="quality-assessment.html"><a href="quality-assessment.html#distribution-of-expression-levels-among-genes"><i class="fa fa-check"></i><b>2.4</b> Distribution of expression levels among genes</a></li>
<li class="chapter" data-level="2.5" data-path="quality-assessment.html"><a href="quality-assessment.html#filtering-of-lowly-expressed-genes"><i class="fa fa-check"></i><b>2.5</b> Filtering of lowly-expressed genes</a></li>
<li class="chapter" data-level="2.6" data-path="quality-assessment.html"><a href="quality-assessment.html#normalization"><i class="fa fa-check"></i><b>2.6</b> Normalization</a></li>
<li class="chapter" data-level="2.7" data-path="quality-assessment.html"><a href="quality-assessment.html#ma-plots"><i class="fa fa-check"></i><b>2.7</b> MA-plots</a></li>
<li class="chapter" data-level="2.8" data-path="quality-assessment.html"><a href="quality-assessment.html#batch-identification"><i class="fa fa-check"></i><b>2.8</b> Batch identification</a></li>
<li class="chapter" data-level="2.9" data-path="quality-assessment.html"><a href="quality-assessment.html#session-information"><i class="fa fa-check"></i><b>2.9</b> Session information</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="quality-assessment" class="section level1">
<h1><span class="header-section-number">2</span> Quality assessment</h1>
<div id="important-variables" class="section level2">
<h2><span class="header-section-number">2.1</span> Important Variables</h2>
<p>In thyroid cancer we consider some variables more important than others as, according to bibliography, significant differences are seen in the categories of this variables related to thyroid cancer. We want to have a first look at these categories before the analysis of the whole data (Figure <a href="quality-assessment.html#fig:numvars">2.1</a>). This will give us an idea of the distribution of our data and can be useful in further analysis.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">3</span>, <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb18-2" data-line-number="2"><span class="co">#Type</span></a>
<a class="sourceLine" id="cb18-3" data-line-number="3">tumor &lt;-<span class="st"> </span>thca[ , <span class="kw">colData</span>(thca)<span class="op">$</span>type <span class="op">==</span><span class="st"> &quot;tumor&quot;</span>]</a>
<a class="sourceLine" id="cb18-4" data-line-number="4">normal &lt;-<span class="st"> </span>thca[ , <span class="kw">colData</span>(thca)<span class="op">$</span>type <span class="op">==</span><span class="st"> &quot;normal&quot;</span>]</a>
<a class="sourceLine" id="cb18-5" data-line-number="5"><span class="kw">dim</span>(tumor)</a></code></pre></div>
<pre><code>[1] 20115   513</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw">dim</span>(normal)</a></code></pre></div>
<pre><code>[1] 20115    59</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1"><span class="kw">barplot</span>(<span class="kw">summary</span>(<span class="kw">colData</span>(thca)<span class="op">$</span>type), <span class="dt">main=</span><span class="st">&quot;Type of sample&quot;</span>, </a>
<a class="sourceLine" id="cb22-2" data-line-number="2">        <span class="dt">ylab=</span><span class="st">&quot;Total counts&quot;</span>, <span class="dt">col =</span> <span class="kw">c</span>(<span class="st">&quot;chocolate1&quot;</span>, <span class="st">&quot;skyblue&quot;</span>), </a>
<a class="sourceLine" id="cb22-3" data-line-number="3">        <span class="dt">cex.main =</span> <span class="dv">4</span>, <span class="dt">cex.lab =</span> <span class="dv">4</span>, <span class="dt">cex.names =</span> <span class="dv">4</span>)</a>
<a class="sourceLine" id="cb22-4" data-line-number="4"><span class="co">#Gender</span></a>
<a class="sourceLine" id="cb22-5" data-line-number="5">male &lt;-<span class="st"> </span>thca[ , <span class="op">!</span><span class="kw">is.na</span>(thca<span class="op">$</span>gender) <span class="op">&amp;</span><span class="st"> </span><span class="kw">colData</span>(thca)<span class="op">$</span>gender <span class="op">==</span><span class="st"> &quot;MALE&quot;</span>]</a>
<a class="sourceLine" id="cb22-6" data-line-number="6">female &lt;-<span class="st"> </span>thca[ , <span class="op">!</span><span class="kw">is.na</span>(thca<span class="op">$</span>gender) <span class="op">&amp;</span><span class="st"> </span><span class="kw">colData</span>(thca)<span class="op">$</span>gender <span class="op">==</span><span class="st"> &quot;FEMALE&quot;</span>]</a>
<a class="sourceLine" id="cb22-7" data-line-number="7"><span class="kw">dim</span>(male)</a></code></pre></div>
<pre><code>[1] 20115   149</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1"><span class="kw">dim</span>(female)</a></code></pre></div>
<pre><code>[1] 20115   406</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1">na &lt;-<span class="st"> </span>thca[ , <span class="kw">is.na</span>(thca<span class="op">$</span>gender)]</a>
<a class="sourceLine" id="cb26-2" data-line-number="2"><span class="kw">dim</span>(na)</a></code></pre></div>
<pre><code>[1] 20115    17</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1"><span class="kw">barplot</span>(<span class="kw">summary</span>(<span class="kw">colData</span>(thca)<span class="op">$</span>gender), <span class="dt">main=</span><span class="st">&quot;Patient&#39;s gender&quot;</span>, </a>
<a class="sourceLine" id="cb28-2" data-line-number="2">        <span class="dt">ylab=</span><span class="st">&quot;Total counts&quot;</span>, <span class="dt">col =</span> <span class="kw">c</span>(<span class="st">&quot;salmon&quot;</span>, <span class="st">&quot;darkolivegreen3&quot;</span>, <span class="st">&quot;darkred&quot;</span>), </a>
<a class="sourceLine" id="cb28-3" data-line-number="3">        <span class="dt">cex.main =</span> <span class="dv">4</span>, <span class="dt">cex.lab =</span> <span class="dv">4</span>, <span class="dt">cex.names =</span> <span class="dv">4</span>)</a>
<a class="sourceLine" id="cb28-4" data-line-number="4"><span class="co">#Age</span></a>
<a class="sourceLine" id="cb28-5" data-line-number="5">colfunc &lt;-<span class="st"> </span><span class="kw">colorRampPalette</span>(<span class="kw">c</span>(<span class="st">&quot;skyblue&quot;</span>, <span class="st">&quot;chocolate1&quot;</span>))</a>
<a class="sourceLine" id="cb28-6" data-line-number="6">colors &lt;-<span class="st"> </span><span class="kw">colfunc</span>(<span class="dv">100</span>)</a>
<a class="sourceLine" id="cb28-7" data-line-number="7"><span class="kw">barplot</span>(<span class="kw">summary</span>(<span class="kw">colData</span>(thca)<span class="op">$</span>age_at_diagnosis), <span class="dt">main=</span><span class="st">&quot;Patient&#39;s age&quot;</span>, </a>
<a class="sourceLine" id="cb28-8" data-line-number="8">        <span class="dt">ylab=</span><span class="st">&quot;Total counts&quot;</span>, <span class="dt">col =</span> colors, <span class="dt">cex.main =</span> <span class="dv">4</span>, </a>
<a class="sourceLine" id="cb28-9" data-line-number="9">        <span class="dt">cex.lab =</span> <span class="dv">4</span>, <span class="dt">cex.names =</span> <span class="dv">4</span>)</a></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:numvars"></span>
<img src="QAanalysis_files/figure-html/numvars-1.png" alt="Number of samples per cathegory. From top to bottom: type, gender and age." width="2400" />
<p class="caption">
Figure 2.1: Number of samples per cathegory. From top to bottom: type, gender and age.
</p>
</div>
<p>The dataset is componed by 513 tumors and 59 normal samples, which can biases our results as it is not balanced. Regarding to the gender, there are 149 males and 406 females samples, with 17 samples from unknown gender, this can be due to that thyroid carcinome is more frequent in females than in males. For that reason, we may would like to analyze the differntial expression analysis between gender, so we would discard samples with NA values. According to the age, we observe there is not significant different among different values, less the last large column which correspond to NA values so we are not taking this factor into account.</p>
<p>Other variables as race, ethnicity, tumor_type and tumor_stage have not been assessed, because, although they are considered important, a big ammount of samples where not defined.</p>
</div>
<div id="sequencing-depth" class="section level2">
<h2><span class="header-section-number">2.2</span> Sequencing depth</h2>
<p>We examine the sequencing depth in terms of total number of sequence read counts mapped to the genome per sample. Figure <a href="quality-assessment.html#fig:libsizes">2.2</a> below shows the
sequencing depth per sample, also known as library sizes, in increasing order.</p>
<div class="figure" style="text-align: center"><span id="fig:libsizes"></span>
<img src="QAanalysis_files/figure-html/libsizes-1.png" alt="Library sizes. Milions of reads per sample in increasing order." width="800px" />
<p class="caption">
Figure 2.2: Library sizes. Milions of reads per sample in increasing order.
</p>
</div>
<p>In this plot, we can observe again that there are more samples from tumor than normal, and both types have similar sequencing depth. However, the figure also reveals substantial differences in sequencing depth between samples, this is why we have to consider discarding those samples whose depth is substantially lower than the rest, by selecting a cutoff greater than 40 million reads.</p>
<p>Moreover, as we have previously reported, this cancer has different affection depending on the gender, so we have to consider performing further analysis comparing females and males. We apply another filter in order to remove those samples with unknown (NA) gender.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1">sampledepth &lt;-<span class="st"> </span><span class="kw">round</span>(dge<span class="op">$</span>sample<span class="op">$</span>lib.size <span class="op">/</span><span class="st"> </span><span class="fl">1e6</span>, <span class="dt">digits=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb29-2" data-line-number="2"><span class="kw">names</span>(sampledepth) &lt;-<span class="st"> </span><span class="kw">substr</span>(<span class="kw">colnames</span>(thca), <span class="dv">6</span>, <span class="dv">12</span>)</a>
<a class="sourceLine" id="cb29-3" data-line-number="3"><span class="kw">sort</span>(sampledepth)</a></code></pre></div>
<pre><code>DJ.A3VJ ET.A3BP ET.A3DO EL.A4K7 EL.A4K2 EL.A3CP EM.A22Q E8.A417 DJ.A13W 
    0.5     2.0     2.2     2.6     3.2    14.4    15.6    16.9    18.0 
EM.A22O KS.A41I EL.A3ZS ET.A25M ET.A25L BJ.A28T EM.A3FQ FK.A4UB FE.A3PD 
   19.2    20.5    20.7    20.9    21.7    22.3    22.9    23.0    23.1 
IM.A41Z DJ.A13R DJ.A1QG KS.A41F EM.A2CK BJ.A45E EM.A1YB ET.A4KN EM.A1CT 
   23.3    23.7    23.7    23.7    24.1    24.9    26.4    26.7    27.4 
EL.A4JX EL.A3ZL BJ.A0ZE KS.A4I7 BJ.A28W DJ.A2PY DJ.A4UR FY.A3R6 DE.A7U5 
   27.5    28.5    29.6    29.6    30.4    30.4    30.5    30.6    31.0 
EM.A1YE EL.A3ZG DJ.A2PQ BJ.A28W EM.A3O3 EM.A22M EM.A2CM DE.A4MA EL.A3H5 
   31.2    31.6    32.7    32.7    32.8    32.9    33.5    33.7    33.7 
BJ.A18Z QD.A8IV EM.A1CW KS.A4I1 FY.A3TY EL.A3ZQ MK.A84Z L6.A4EP DE.A4M8 
   34.0    34.1    34.3    34.4    34.8    34.9    34.9    35.3    35.4 
ET.A25R GE.A2C6 ET.A3DR FE.A230 DJ.A2QA EM.A4G1 FY.A76V KS.A4ID EM.A4FM 
   35.6    35.6    35.7    35.8    36.0    36.2    36.2    36.2    36.6 
BJ.A0Z9 EM.A3SZ ET.A25K EL.A3CW DE.A4MB DJ.A2Q0 FE.A22Z FE.A3PC KS.A4IB 
   36.9    36.9    36.9    37.0    37.1    37.4    37.5    37.7    37.7 
E8.A437 DJ.A4UT EL.A4K1 EM.A3ST E8.A2JQ BJ.A28S E8.A3X7 EM.A3O9 KS.A4I3 
   38.0    38.1    38.2    38.5    38.6    38.7    38.9    39.0    39.3 
BJ.A191 EM.A1YC BJ.A0Z2 BJ.A45F EM.A3SU ET.A40S E8.A44K EL.A3D5 DJ.A4V4 
   39.4    39.4    39.6    39.6    39.6    39.6    39.7    39.9    40.0 
FY.A3YR EM.A2CO EM.A4FV ET.A39M BJ.A2NA DJ.A2PZ BJ.A0Z5 BJ.A45H EM.A2OY 
   40.4    40.5    40.6    40.7    40.7    40.8    40.9    41.2    41.2 
ET.A3DU KS.A4I9 DJ.A3VL L6.A4EQ EL.A4K9 BJ.A2P4 EM.A3FN IM.A4EB FY.A3TY 
   41.2    41.2    41.3    41.5    41.6    41.7    41.7    41.8    41.8 
4C.A93U DJ.A13V E8.A433 EM.A3ST BJ.A3PT EM.A3SY KS.A41I BJ.A0ZJ ET.A3DQ 
   41.9    42.0    42.0    42.2    42.3    42.4    42.4    42.6    42.6 
DE.A4MD ET.A39R J8.A4HW KS.A4I5 EM.A2P1 KS.A4IC IM.A41Y KS.A41L FY.A3BL 
   42.7    42.8    42.9    42.9    43.0    43.0    43.1    43.1    43.2 
FY.A4B0 EL.A3D4 MK.A4N9 EL.A3CM DJ.A1QO J8.A4HY DJ.A3V0 EM.A2CP BJ.A28Z 
   43.3    43.6    43.6    43.8    44.0    44.0    44.1    44.1    44.2 
DE.A69K DJ.A3V6 EM.A2P1 BJ.A4O8 DE.A4M9 EL.A3N3 EL.A3ZN EM.A3AO BJ.A0YZ 
   44.4    44.4    44.4    44.5    44.5    44.6    44.6    44.6    44.7 
E8.A44M BJ.A3PR DE.A0XZ BJ.A3PR CE.A13K DJ.A3UW EL.A3GO EL.A3ZT EL.A4KI 
   44.7    44.8    44.8    44.8    44.9    44.9    44.9    44.9    44.9 
EM.A3AJ EM.A2OW EM.A1CU DJ.A2QC EL.A3CU BJ.A0ZC CE.A3MD DJ.A4UL BJ.A45I 
   44.9    45.0    45.0    45.1    45.1    45.2    45.2    45.2    45.3 
EL.A3CX E8.A2JQ EL.A3ZM DJ.A13M E3.A3E0 ET.A2N3 E8.A438 EL.A3CL J8.A3YH 
   45.3    45.3    45.4    45.5    45.5    45.7    45.9    45.9    45.9 
EM.A4FR EL.A3GZ DJ.A2PS EL.A3GR EM.A4FN EM.A3FP ET.A39T DJ.A3VM EL.A4K6 
   46.0    46.0    46.3    46.4    46.4    46.5    46.5    46.6    46.6 
EM.A2OV E8.A2EA EM.A1YD ET.A2N0 FY.A40N BJ.A291 EL.A4JV ET.A2N5 BJ.A45C 
   46.6    46.7    46.7    46.7    46.7    46.8    46.8    46.8    46.9 
ET.A3BW E3.A3E5 BJ.A28R EL.A3H8 EL.A3GP EL.A3T1 DO.A1K0 EL.A3H3 EM.A4FO 
   46.9    47.0    47.1    47.2    47.3    47.3    47.4    47.4    47.4 
FY.A3NN J8.A3YD EL.A3ZP CE.A485 EL.A3T3 DJ.A3VK ET.A2MX BJ.A0ZG BJ.A3PU 
   47.4    47.4    47.5    47.7    47.7    47.8    47.8    47.9    47.9 
EL.A3ZP ET.A3DT EL.A3ZK ET.A40P BJ.A0Z0 DE.A0Y3 EM.A2CR ET.A39N CE.A3ME 
   47.9    47.9    48.0    48.1    48.2    48.2    48.2    48.2    48.3 
DJ.A4UW EL.A3TA EL.A3ZS EM.A3AQ ET.A2MY EM.A1YC L6.A4ET EM.A2OX EM.A3AR 
   48.3    48.3    48.3    48.3    48.3    48.4    48.4    48.5    48.5 
DJ.A3VE EL.A3GU CE.A484 DJ.A2PT DO.A1JZ EL.A3ZK ET.A2MX E3.A3E2 EL.A3T8 
   48.6    48.6    48.7    48.7    48.7    48.9    48.9    49.0    49.0 
FY.A2QD DJ.A4UQ FY.A40K EL.A3ZO DJ.A4UP DJ.A4V0 E8.A419 EL.A3D0 EL.A3D6 
   49.0    49.1    49.1    49.1    49.2    49.2    49.2    49.2    49.2 
EL.A3D1 EL.A3GQ J8.A42S ET.A39O J8.A4HW BJ.A192 DJ.A2QB DJ.A3VD E3.A3DZ 
   49.3    49.3    49.3    49.4    49.4    49.5    49.6    49.6    49.6 
DJ.A2Q1 DJ.A3UU E8.A413 EL.A3T0 EL.A3CN EL.A3CS ET.A40T DE.A4MD EM.A2CN 
   49.7    49.7    49.7    49.7    49.8    49.8    49.8    49.9    49.9 
DO.A1JZ BJ.A28X EL.A4JZ EM.A1CV BJ.A0ZA DJ.A1QL DJ.A2PO EM.A22I CE.A481 
   50.0    50.1    50.1    50.1    50.2    50.3    50.3    50.3    50.4 
EL.A3MX EL.A4K0 EL.A4JW ET.A39K ET.A3BV EL.A4KH ET.A2N4 EM.A1CS BJ.A2N9 
   50.4    50.4    50.5    50.5    50.5    50.6    50.6    50.6    50.7 
BJ.A18Y DO.A2HM EM.A1CU BJ.A28X BJ.A45G DJ.A4V5 EM.A2P3 ET.A39P EM.A2P0 
   50.8    50.8    50.8    50.9    51.0    51.0    51.0    51.0    51.1 
J8.A3O2 ET.A3DW DJ.A3V3 EL.A3CO EM.A3SX BJ.A2N9 ET.A25N DJ.A4V2 EM.A2CJ 
   51.1    51.1    51.2    51.2    51.2    51.2    51.3    51.4    51.4 
BJ.A3F0 EM.A3AI FY.A3I5 DJ.A13U EL.A3CZ EL.A3GZ FE.A232 EL.A3T8 EL.A3ZR 
   51.5    51.5    51.5    51.6    51.7    51.8    51.8    51.8    51.9 
J8.A3YF EL.A3ZT DJ.A3UZ EM.A22P DE.A0Y2 EL.A3H2 EM.A4FK ET.A39S BJ.A45K 
   51.9    51.9    52.1    52.3    52.4    52.4    52.4    52.4    52.5 
DJ.A13L EL.A3CY FY.A3RA DJ.A3UK H2.A422 J8.A3O2 E3.A3E3 EL.A3T0 ET.A3BO 
   52.5    52.6    52.6    52.7    52.7    52.7    52.9    53.0    53.0 
J8.A3YE ET.A39J ET.A3DW MK.A4N6 DJ.A1QQ EM.A3FO ET.A25G FE.A3PA FY.A40M 
   53.0    53.1    53.1    53.1    53.2    53.2    53.2    53.2    53.2 
EL.A3GV EL.A3MW EL.A3H2 DJ.A3UR EM.A2CS FE.A235 EL.A3MW ET.A3BX BJ.A0ZH 
   53.3    53.3    53.4    53.6    53.6    53.6    53.6    53.7    53.8 
DJ.A3UN EL.A3MY EM.A3FQ FY.A4B4 IM.A420 DE.A3KN DJ.A2PW EL.A3T2 EL.A3CT 
   53.8    53.9    53.9    54.0    54.0    54.1    54.1    54.3    54.4 
EL.A3T6 EL.A3CR EM.A22J FE.A237 BJ.A45D BJ.A45J EM.A3FR FY.A40L ET.A3BN 
   54.4    54.5    54.5    54.5    54.6    54.6    54.6    54.6    54.7 
CE.A482 DJ.A3VI EM.A1YA EM.A2CS BJ.A190 EL.A3TB EL.A3T9 EM.A3FL KS.A41L 
   54.8    54.8    54.8    54.8    54.9    54.9    55.0    55.0    55.0 
EL.A4KD FY.A3ON J8.A3O0 DJ.A3VG EL.A3GX EM.A22K EM.A3O7 E8.A432 E8.A436 
   55.1    55.1    55.1    55.2    55.2    55.2    55.2    55.3    55.3 
J8.A3YH EL.A3H7 EL.A3ZR BJ.A290 DJ.A3UP EL.A3GS DJ.A2PR DJ.A2PV EL.A3N2 
   55.3    55.3    55.4    55.5    55.5    55.5    55.6    55.6    55.6 
KS.A41J DJ.A3UO EL.A3H4 DE.A4MC DJ.A3US FK.A3S3 DJ.A3UQ E8.A434 FE.A3PB 
   55.7    55.8    55.8    55.9    55.9    55.9    56.0    56.1    56.1 
FY.A4B3 DJ.A2PN H2.A421 IM.A3U2 EM.A1CW EM.A2CU DJ.A2Q9 EL.A3ZH EM.A4FQ 
   56.1    56.2    56.2    56.2    56.3    56.3    56.5    56.5    56.5 
EM.A1CT EL.A3CV ET.A3BS DJ.A3UY ET.A25J CE.A483 EL.A3MZ FE.A239 EL.A3MY 
   56.6    56.7    56.7    56.8    56.8    56.9    56.9    56.9    56.9 
DJ.A3UX E8.A416 EL.A3T7 ET.A25O ET.A3BU EM.A3FK ET.A3DS GE.A2C6 DJ.A2PX 
   57.0    57.0    57.0    57.2    57.2    57.3    57.3    57.3    57.4 
DJ.A1QE DJ.A3UT EL.A3H1 EM.A3FM EL.A3ZL DJ.A1QI EL.A4K4 BJ.A2NA DJ.A13S 
   57.5    57.5    57.5    57.7    57.7    57.8    57.9    58.0    58.3 
EM.A3OB BJ.A2N8 FY.A3R9 EL.A3T3 EL.A3GW IM.A3ED ET.A40R EM.A2P2 DJ.A13P 
   58.3    58.5    58.5    58.5    58.6    58.6    58.7    58.8    58.9 
DJ.A13X DJ.A3V5 BJ.A0ZB DJ.A2Q2 EL.A3ZG DJ.A3VA EL.A3N3 FY.A3R8 ET.A2N5 
   58.9    58.9    59.0    59.0    59.0    59.1    59.1    59.2    59.2 
DJ.A2PU FK.A3SH DJ.A3UV EM.A3SU EM.A4FU H2.A3RH J8.A3NZ DJ.A3V4 EL.A3TB 
   59.3    59.3    59.4    59.4    59.4    59.4    59.4    59.7    59.7 
DJ.A3V7 DJ.A3V8 EL.A3ZM J8.A3O1 ET.A39L H2.A3RI EM.A4FH DJ.A2Q5 ET.A25P 
   59.8    59.8    59.8    59.8    59.9    60.1    60.2    60.3    60.4 
BJ.A4O9 EM.A3FJ EL.A3T2 E3.A3DY EL.A3T6 EL.A3ZH E3.A3E1 EM.A1CS ET.A3DV 
   60.5    60.5    60.8    60.9    61.0    61.0    61.1    61.1    61.1 
E8.A414 ET.A39I FY.A3R7 H2.A26U IM.A3EB DJ.A13T DJ.A2PP EM.A4FF MK.A4N7 
   61.2    61.3    61.3    61.3    61.4    61.6    61.6    61.7    61.8 
EL.A3TA IM.A3U3 EL.A3ZO FK.A3SG DJ.A3V2 ET.A3BT EL.A3H7 EM.A22N FK.A3SD 
   61.8    62.0    62.1    62.3    62.5    62.6    62.8    62.8    62.9 
BJ.A3EZ BJ.A28V FK.A3SB H2.A2K9 DJ.A3VF FE.A238 EM.A3AK BJ.A3PU BJ.A0ZF 
   63.0    63.1    63.1    63.2    63.3    63.3    63.5    63.5    63.6 
EL.A3N2 EL.A4KG EL.A3H1 EM.A3AL EM.A3AN FY.A3NP DJ.A3V9 ET.A3DP EM.A2CT 
   63.7    63.8    63.8    64.1    64.2    64.7    64.9    65.0    65.3 
FE.A23A FE.A236 BJ.A2N7 ET.A3DP E8.A415 EM.A2CL FE.A234 KS.A41J BJ.A2N7 
   65.3    65.4    65.4    65.5    65.8    65.8    65.8    65.9    66.2 
FY.A3WA EM.A2CQ H2.A3RI EL.A3ZQ ET.A2MZ J8.A3YG EL.A3T1 ET.A25I E8.A418 
   66.3    66.4    66.4    66.5    67.8    68.1    68.2    68.9    69.0 
EM.A3O6 EM.A1CV FY.A3W9 L6.A4EU BJ.A290 ET.A3BQ DJ.A2Q3 DJ.A3UM BJ.A0Z3 
   69.1    69.2    69.4    69.4    69.4    69.7    69.8    69.9    70.4 
EM.A3O8 DJ.A1QH EL.A3MX ET.A40Q FY.A3I4 FK.A3SE ET.A4KQ FE.A231 DE.A69J 
   70.8    71.4    72.0    72.3    72.3    72.4    72.6    72.8    73.0 
DJ.A1QD BJ.A28R CE.A27D EM.A3AP DJ.A2Q7 EL.A3GY E8.A242 DE.A2OL DJ.A2Q6 
   73.4    73.5    73.5    73.5    73.6    73.6    74.6    75.0    75.5 
DJ.A1QM DJ.A2Q4 DJ.A3VB EL.A3T7 EM.A22L FY.A3NM DJ.A13O EM.A2OZ DJ.A1QN 
   75.8    76.6    76.6    77.2    78.9    79.1    82.9    84.0    85.7 
DJ.A1QF BJ.A2N8 H2.A2K9 EM.A3OA FE.A233 
   86.2    89.1    91.1    95.4   104.0 </code></pre>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1"><span class="co">#sample depth filter</span></a>
<a class="sourceLine" id="cb31-2" data-line-number="2">maskd &lt;-<span class="st"> </span>sampledepth <span class="op">&gt;</span><span class="st"> </span><span class="dv">40</span></a>
<a class="sourceLine" id="cb31-3" data-line-number="3"><span class="co">#gender filter</span></a>
<a class="sourceLine" id="cb31-4" data-line-number="4">maskg &lt;-<span class="st"> </span><span class="op">!</span><span class="kw">is.na</span>(thca<span class="op">$</span>gender)</a>
<a class="sourceLine" id="cb31-5" data-line-number="5">thca.filt &lt;-<span class="st"> </span>thca[, maskd<span class="op">&amp;</span>maskg]</a>
<a class="sourceLine" id="cb31-6" data-line-number="6">dge.filt &lt;-<span class="st"> </span>dge[, maskd<span class="op">&amp;</span>maskg]</a>
<a class="sourceLine" id="cb31-7" data-line-number="7"><span class="kw">dim</span>(thca)</a></code></pre></div>
<pre><code>[1] 20115   572</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1"><span class="kw">dim</span>(thca.filt)</a></code></pre></div>
<pre><code>[1] 20115   467</code></pre>
<p>Once the filters are applied, the dataset have decreased to 467. Now, we perform another sequencing depth distribution (Figure <a href="quality-assessment.html#fig:filtlibsizes">2.3</a>) in order to check the effect of the filter.</p>
<div class="figure" style="text-align: center"><span id="fig:filtlibsizes"></span>
<img src="QAanalysis_files/figure-html/filtlibsizes-1.png" alt="Library size. Milions of reads per sample in increasing order after filtering." width="800px" />
<p class="caption">
Figure 2.3: Library size. Milions of reads per sample in increasing order after filtering.
</p>
</div>
</div>
<div id="distribution-of-expression-levels-among-samples" class="section level2">
<h2><span class="header-section-number">2.3</span> Distribution of expression levels among samples</h2>
<p>One way to normalize RNA-seq data is an adjustment to compare across features
in a sample, this can be performed using Count Per Million (CPM) reads or its logarithm. The distribution of expression values per sample in terms of logarithmic CPM units is ploted in Figure <a href="quality-assessment.html#fig:distRawExp">2.4</a> separating by tumor and normal samples.</p>
<p>A box plot of the expression values per samples is also performed (Figure <a href="quality-assessment.html#fig:distboxplot">2.5</a>) in order to have another visual way to interpret the data.</p>
<div class="figure" style="text-align: center"><span id="fig:distRawExp"></span>
<img src="QAanalysis_files/figure-html/distRawExp-1.png" alt="Non-parametric density distribution of expression profiles per sample. Shows density of genes per log2CPM. Tumor samples are ploted in the left and normal samples in the right." width="800px" />
<p class="caption">
Figure 2.4: Non-parametric density distribution of expression profiles per sample. Shows density of genes per log2CPM. Tumor samples are ploted in the left and normal samples in the right.
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:distboxplot"></span>
<img src="QAanalysis_files/figure-html/distboxplot-1.png" alt="Box plot. Distribution of the expression values across samples." width="800px" />
<p class="caption">
Figure 2.5: Box plot. Distribution of the expression values across samples.
</p>
</div>
<p>Multidensity plots show that all samples follow a similar expression pattern, with a group of one of lowly expressed genes and another highly expressed ones, being similar between tumor and normal. The box plot shows that there is no sample with a high deviation from the average interquartile range.</p>
<p>For all this reasons, we assume that we do not need further normalization among samples.</p>
</div>
<div id="distribution-of-expression-levels-among-genes" class="section level2">
<h2><span class="header-section-number">2.4</span> Distribution of expression levels among genes</h2>
<p>In order to identify lowly expressed genes, the average expression per gene through all samples is calculated. The distribution of those values across genes is represented in the Figure <a href="quality-assessment.html#fig:exprdist">2.6</a>.</p>
<div class="figure" style="text-align: center"><span id="fig:exprdist"></span>
<img src="QAanalysis_files/figure-html/exprdist-1.png" alt="Distribution of expression level. Shows the frequency of each gene per average expression level." width="800px" />
<p class="caption">
Figure 2.6: Distribution of expression level. Shows the frequency of each gene per average expression level.
</p>
</div>
<p>As we have seen in the multidensity plot, there is a region of lowly expressed genes <em>(<span class="math inline">\(\log_2\)</span>CPM &lt; 0)</em> and other with highly expressed ones <em>(<span class="math inline">\(\log_2\)</span>CPM &gt; 0)</em>, which will be analyzed further on.</p>
</div>
<div id="filtering-of-lowly-expressed-genes" class="section level2">
<h2><span class="header-section-number">2.5</span> Filtering of lowly-expressed genes</h2>
<p>Lowly-expressed genes can be filtered following two criteria:</p>
<ul>
<li><p>Filtering out genes below a minimum average <span class="math inline">\(\log_2\)</span>CPM throught the samples.</p></li>
<li><p>Filtering out genes with fewer than a given number of sample meeting a minimum <span class="math inline">\(\log_2\)</span>CPM</p></li>
</ul>
<p>First of all, we are going to filter genes using the second approach. An histograme is plotted in Figure <a href="quality-assessment.html#fig:expfilt1">2.7</a> in order to visualize the effect of the filtering.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1"><span class="co">#calculate cpm cutoff around all samples</span></a>
<a class="sourceLine" id="cb35-2" data-line-number="2">cpmcutoff &lt;-<span class="st"> </span><span class="kw">round</span>(<span class="dv">10</span><span class="op">/</span><span class="kw">min</span>(dge.filt<span class="op">$</span>sample<span class="op">$</span>lib.size<span class="op">/</span><span class="fl">1e+06</span>), <span class="dt">digits =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb35-3" data-line-number="3">cpmcutoff</a></code></pre></div>
<pre><code>[1] 0.2</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb37-1" data-line-number="1"><span class="co">#select number of samples meeting that cutoff</span></a>
<a class="sourceLine" id="cb37-2" data-line-number="2">nsamplescutoff &lt;-<span class="st"> </span><span class="kw">min</span>(<span class="kw">table</span>(thca.filt<span class="op">$</span>type))</a>
<a class="sourceLine" id="cb37-3" data-line-number="3">nsamplescutoff</a></code></pre></div>
<pre><code>[1] 53</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb39-1" data-line-number="1">mask &lt;-<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">cpm</span>(dge.filt) <span class="op">&gt;</span><span class="st"> </span>cpmcutoff) <span class="op">&gt;=</span><span class="st"> </span>nsamplescutoff</a>
<a class="sourceLine" id="cb39-2" data-line-number="2">thca.filt2 &lt;-<span class="st"> </span>thca.filt[mask, ]</a>
<a class="sourceLine" id="cb39-3" data-line-number="3">dge.filt2 &lt;-<span class="st"> </span>dge.filt[mask, ]</a>
<a class="sourceLine" id="cb39-4" data-line-number="4"><span class="kw">dim</span>(thca)</a></code></pre></div>
<pre><code>[1] 20115   572</code></pre>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb41-1" data-line-number="1"><span class="kw">dim</span>(thca.filt2)</a></code></pre></div>
<pre><code>[1] 15408   467</code></pre>
<div class="figure" style="text-align: center"><span id="fig:expfilt1"></span>
<img src="QAanalysis_files/figure-html/expfilt1-1.png" alt="Distribution of average expression level per gene. Filtering out genes with `r nsamplescutoff` samples with a log2CPM lower than `r cpmcutoff`. Number of genes passing the filter are shown in red." width="800px" />
<p class="caption">
Figure 2.7: Distribution of average expression level per gene. Filtering out genes with <code>r nsamplescutoff</code> samples with a log2CPM lower than <code>r cpmcutoff</code>. Number of genes passing the filter are shown in red.
</p>
</div>
<p>The dataset has been reduced to 15408 genes, which means that a lot of genes have passed the cuttoff (red bars). As we still have a lot of lowly expressed genes, we apply a second filter following the first approach, by choosing a cut-off of <span class="math inline">\(\log_2\)</span>CPM = 3, and plot another histogram (Figure <a href="quality-assessment.html#fig:expfilt2">2.8</a>) in order to visualize the results.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb43-1" data-line-number="1">mask &lt;-<span class="st"> </span>avgexp <span class="op">&gt;</span><span class="st"> </span><span class="dv">3</span></a>
<a class="sourceLine" id="cb43-2" data-line-number="2">thca.filt3 &lt;-<span class="st"> </span>thca.filt[mask, ]</a>
<a class="sourceLine" id="cb43-3" data-line-number="3">dge.filt3 &lt;-<span class="st"> </span>dge.filt[mask, ]</a>
<a class="sourceLine" id="cb43-4" data-line-number="4"><span class="kw">dim</span>(thca.filt3)</a></code></pre></div>
<pre><code>[1] 9423  467</code></pre>
<div class="figure" style="text-align: center"><span id="fig:expfilt2"></span>
<img src="QAanalysis_files/figure-html/expfilt2-1.png" alt="Distribution of average expression level per gene with a minimum log2CPM of 3. Numbers of genes passing the previous filter are shown in red. Number of genes passing the current filter are shown in blue" width="800px" />
<p class="caption">
Figure 2.8: Distribution of average expression level per gene with a minimum log2CPM of 3. Numbers of genes passing the previous filter are shown in red. Number of genes passing the current filter are shown in blue
</p>
</div>
<p>We observe that the second filter is more restrictive, having a total of 9423 final genes, so we are going to use this filtered dataset for further analysis.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb45-1" data-line-number="1"><span class="kw">saveRDS</span>(thca.filt3, <span class="kw">file.path</span>(<span class="st">&quot;results&quot;</span>, <span class="st">&quot;thca.filt.unnorm.rds&quot;</span>))</a>
<a class="sourceLine" id="cb45-2" data-line-number="2"><span class="kw">saveRDS</span>(dge.filt3, <span class="kw">file.path</span>(<span class="st">&quot;results&quot;</span>, <span class="st">&quot;dge.filt.unnorm.rds&quot;</span>))</a></code></pre></div>
</div>
<div id="normalization" class="section level2">
<h2><span class="header-section-number">2.6</span> Normalization</h2>
<p>After filtering for lowly expressed genes, normalization factors need to be calculated in order to allow the detection of differentialy expressed genes due to the interesting outcome and not to other unknown factors. Furthermore, we need to store the normalized <span class="math inline">\(\log_2\)</span>CPM values into the filtered dataset and store normalized version for the filtered data.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb46-1" data-line-number="1">dge.filt3 &lt;-<span class="st"> </span><span class="kw">calcNormFactors</span>(dge.filt3)</a>
<a class="sourceLine" id="cb46-2" data-line-number="2"><span class="kw">head</span>(dge.filt3<span class="op">$</span>samples<span class="op">$</span>norm.factors)</a></code></pre></div>
<pre><code>[1] 0.8471913 0.9717390 0.9386724 1.0353601 1.0518953 0.9920039</code></pre>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb48-1" data-line-number="1"><span class="kw">assays</span>(thca.filt3)<span class="op">$</span>logCPM &lt;-<span class="st"> </span><span class="kw">cpm</span>(dge.filt3, <span class="dt">log=</span><span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb48-2" data-line-number="2">                                 <span class="dt">normalized.lib.sizes=</span><span class="ot">TRUE</span>, <span class="dt">prior.count=</span><span class="fl">0.25</span>)</a></code></pre></div>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb49-1" data-line-number="1"><span class="kw">saveRDS</span>(thca.filt3, <span class="kw">file.path</span>(<span class="st">&quot;results&quot;</span>, <span class="st">&quot;thca.filt.rds&quot;</span>))</a>
<a class="sourceLine" id="cb49-2" data-line-number="2"><span class="kw">saveRDS</span>(dge.filt3, <span class="kw">file.path</span>(<span class="st">&quot;results&quot;</span>, <span class="st">&quot;dge.filt.rds&quot;</span>))</a></code></pre></div>
</div>
<div id="ma-plots" class="section level2">
<h2><span class="header-section-number">2.7</span> MA-plots</h2>
<p>MA plots are used to detect intensity dependent biases, by comparing two groups of the dataset. Firstly, a global MA-plot is examined for normalized and non-normalized (Figure <a href="quality-assessment.html#fig:maPlotsGeneral">2.9</a>) and non-normalized data (Figure <a href="quality-assessment.html#fig:maPlotsGeneral2">2.10</a>)</p>
<pre><code>
normal  tumor 
    59    513 </code></pre>
<div class="figure" style="text-align: center"><span id="fig:maPlotsGeneral"></span>
<img src="QAanalysis_files/figure-html/maPlotsGeneral-1.png" alt="MA-plots of normalized data. Shows fold change in logarithmic scale in function of the average expression." width="800px" />
<p class="caption">
Figure 2.9: MA-plots of normalized data. Shows fold change in logarithmic scale in function of the average expression.
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:maPlotsGeneral2"></span>
<img src="QAanalysis_files/figure-html/maPlotsGeneral2-1.png" alt="MA-plots of non-normalized data. Shows fold change (FC) in logarithmic scale in function of the average expression." width="800px" />
<p class="caption">
Figure 2.10: MA-plots of non-normalized data. Shows fold change (FC) in logarithmic scale in function of the average expression.
</p>
</div>
<p>In the non-normalized plot, we can observe there are lots of genes non differentialy expreesed (logFC ~ 0) and also several ones with a low <span class="math inline">\(\log_2\)</span>CPM. Those lowly expressed genes ones are removed in the normalized plot in order to avoid artifacts in posterior analysis, so it presents a smother shape with less outliers.</p>
<p>MA-plots of the normalized expression profiles for each tumor (Figure <a href="quality-assessment.html#fig:maPlotsTumor">2.11</a>) and normal (Figure <a href="quality-assessment.html#fig:maPlotsNormal">2.12</a>) samples sparately are then examined to look for anomalous expression profiles.</p>
<div class="figure" style="text-align: center"><span id="fig:maPlotsTumor"></span>
<img src="QAanalysis_files/figure-html/maPlotsTumor-1.png" alt="MA-plots for each tumor sample. Show fold change in logarithmic scale in function of the average expression" width="1000" />
<p class="caption">
Figure 2.11: MA-plots for each tumor sample. Show fold change in logarithmic scale in function of the average expression
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:maPlotsTumor"></span>
<img src="QAanalysis_files/figure-html/maPlotsTumor-2.png" alt="MA-plots for each tumor sample. Show fold change in logarithmic scale in function of the average expression" width="1000" />
<p class="caption">
Figure 2.11: MA-plots for each tumor sample. Show fold change in logarithmic scale in function of the average expression
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:maPlotsTumor"></span>
<img src="QAanalysis_files/figure-html/maPlotsTumor-3.png" alt="MA-plots for each tumor sample. Show fold change in logarithmic scale in function of the average expression" width="1000" />
<p class="caption">
Figure 2.11: MA-plots for each tumor sample. Show fold change in logarithmic scale in function of the average expression
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:maPlotsTumor"></span>
<img src="QAanalysis_files/figure-html/maPlotsTumor-4.png" alt="MA-plots for each tumor sample. Show fold change in logarithmic scale in function of the average expression" width="1000" />
<p class="caption">
Figure 2.11: MA-plots for each tumor sample. Show fold change in logarithmic scale in function of the average expression
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:maPlotsTumor"></span>
<img src="QAanalysis_files/figure-html/maPlotsTumor-5.png" alt="MA-plots for each tumor sample. Show fold change in logarithmic scale in function of the average expression" width="1000" />
<p class="caption">
Figure 2.11: MA-plots for each tumor sample. Show fold change in logarithmic scale in function of the average expression
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:maPlotsNormal"></span>
<img src="QAanalysis_files/figure-html/maPlotsNormal-1.png" alt="MA-plots for each normal sample. Show fold change in logarithmic scale in function of the average expression." width="1000" />
<p class="caption">
Figure 2.12: MA-plots for each normal sample. Show fold change in logarithmic scale in function of the average expression.
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:maPlotsNormal"></span>
<img src="QAanalysis_files/figure-html/maPlotsNormal-2.png" alt="MA-plots for each normal sample. Show fold change in logarithmic scale in function of the average expression." width="1000" />
<p class="caption">
Figure 2.12: MA-plots for each normal sample. Show fold change in logarithmic scale in function of the average expression.
</p>
</div>
<p>On one hand, most of tumor samples have a uniform distribution, except for the samples ET-A2N0, EM-A3FM and IM-A41Y, where we can observe an important deviation at the end.</p>
<p>On the other hand, several normal samples have deviations, such as BJ-A28X, EL-A3ZP, ET-A3DP and KS-A41I. All those samples are going to be removed in order to avoid biasses in the results. We will store these datasets as the final filtered data.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb51-1" data-line-number="1">maskbad &lt;-<span class="st"> </span><span class="kw">substr</span>(<span class="kw">colnames</span>(thca.filt3), <span class="dv">0</span>, <span class="dv">12</span>) <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb51-2" data-line-number="2"><span class="st">  </span><span class="kw">c</span>(<span class="st">&quot;TCGA.BJ.A28X&quot;</span>, <span class="st">&quot;TGCA.EL.A3ZP&quot;</span>, <span class="st">&quot;TCGA.ET.A3DP&quot;</span>,<span class="st">&quot;TCGA.KS.A41I&quot;</span>,</a>
<a class="sourceLine" id="cb51-3" data-line-number="3">    <span class="st">&quot;TCGA.ET.A2N0&quot;</span>, <span class="st">&quot;TCGA.EM.A3FM&quot;</span>, <span class="st">&quot;TCGA.IM.A41Y&quot;</span>)</a>
<a class="sourceLine" id="cb51-4" data-line-number="4">thca.filt4 &lt;-<span class="st"> </span>thca.filt3[, <span class="op">!</span>maskbad]</a>
<a class="sourceLine" id="cb51-5" data-line-number="5">dge.filt4 &lt;-<span class="st"> </span>dge.filt3[, <span class="op">!</span>maskbad]</a>
<a class="sourceLine" id="cb51-6" data-line-number="6"><span class="kw">dim</span>(thca.filt4)</a></code></pre></div>
<pre><code>[1] 9423  459</code></pre>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb53-1" data-line-number="1"><span class="co">#Save new filtered datasets</span></a>
<a class="sourceLine" id="cb53-2" data-line-number="2"><span class="kw">saveRDS</span>(thca.filt4, <span class="kw">file.path</span>(<span class="st">&quot;results&quot;</span>, <span class="st">&quot;thca.filt4.rds&quot;</span>))</a>
<a class="sourceLine" id="cb53-3" data-line-number="3"><span class="kw">saveRDS</span>(dge.filt4, <span class="kw">file.path</span>(<span class="st">&quot;results&quot;</span>, <span class="st">&quot;dge.filt4.rds&quot;</span>))</a></code></pre></div>
<p>Once normalization is done, a Multidimensinal Scaling (MDS) plot (Figure <a href="quality-assessment.html#fig:MDS">2.13</a>), is performed in order to visualize possible anamoalous samples or clusters with distinctive features.</p>
<div class="figure" style="text-align: center"><span id="fig:MDS"></span>
<img src="QAanalysis_files/figure-html/MDS-1.png" alt="MDS plot. Shows tumor (blue) and normal (orange) samples grouped by fold change in logarithmic scale" width="800px" />
<p class="caption">
Figure 2.13: MDS plot. Shows tumor (blue) and normal (orange) samples grouped by fold change in logarithmic scale
</p>
</div>
<p>The MDS plot shows two possible clusters, which highlights a difference between normal and tummor samples. However, a test is needed to asses if the cluster occures due to the type of tissue or because there is any batch effect that confounds our outcome of interest; in order to avoid missundertandments in further analysis.</p>
<p>Moreover, in the normal samples cluster we can find several tumor samples. If we ignore the batch effect, leading that the clustering is due to sample type, this fact can be explained becuse not all thyroid tumors have the same mutations, which produces different levels of severity in cancer patients.</p>
<p>As we have previously reported, this type of cancer can have different incidence regarding to the gender of the patient. That way, we also perform a MDS plot (Figure <a href="quality-assessment.html#fig:MDS2">2.14</a>) of female and male samples in order to observe possible differential expression pattern.</p>
<div class="figure" style="text-align: center"><span id="fig:MDS2"></span>
<img src="QAanalysis_files/figure-html/MDS2-1.png" alt="MDS of the samples according to its gender (male or female)." width="800px" />
<p class="caption">
Figure 2.14: MDS of the samples according to its gender (male or female).
</p>
</div>
<p>This plot shows that there is not a cluster that clearly divides both groups, so we decide not to continue analyzing differential expressed genes within gender.</p>
</div>
<div id="batch-identification" class="section level2">
<h2><span class="header-section-number">2.8</span> Batch identification</h2>
<p>The next step is to search for potential surrogate variables. Given that each sample
name corresponds to a TCGA barcode, different elements are going to be derived of this barcode, and their destribution will be examined across samples.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb54-1" data-line-number="1">tss &lt;-<span class="st"> </span><span class="kw">substr</span>(<span class="kw">colnames</span>(thca.filt4), <span class="dv">6</span>, <span class="dv">7</span>)</a>
<a class="sourceLine" id="cb54-2" data-line-number="2"><span class="kw">table</span>(tss)</a></code></pre></div>
<pre><code>tss
4C BJ CE DE DJ DO E3 E8 EL EM ET FE FK FY GE H2 IM J8 KS L6 MK 
 1 46  9  9 80  4  7 15 97 67 46 12  6 23  1  8  6  9  7  3  3 </code></pre>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb56-1" data-line-number="1">center &lt;-<span class="st"> </span><span class="kw">substr</span>(<span class="kw">colnames</span>(thca.filt4), <span class="dv">27</span>, <span class="dv">28</span>)</a>
<a class="sourceLine" id="cb56-2" data-line-number="2"><span class="kw">table</span>(center)</a></code></pre></div>
<pre><code>center
 07 
459 </code></pre>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb58-1" data-line-number="1">plate &lt;-<span class="st"> </span><span class="kw">substr</span>(<span class="kw">colnames</span>(thca.filt4), <span class="dv">22</span>, <span class="dv">25</span>)</a>
<a class="sourceLine" id="cb58-2" data-line-number="2"><span class="kw">table</span>(plate)</a></code></pre></div>
<pre><code>plate
A10U A13Y A14Y A16R A180 A18C A19O A206 A20F A21D A220 A22L A22U A23N A23W 
  20   16   17   20   19   22   55   34   26   24   25   42   14   61   18 
A250 A39I 
  43    3 </code></pre>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb60-1" data-line-number="1">portionanalyte &lt;-<span class="st"> </span><span class="kw">substr</span>(<span class="kw">colnames</span>(thca.filt4), <span class="dv">18</span>, <span class="dv">20</span>)</a>
<a class="sourceLine" id="cb60-2" data-line-number="2"><span class="kw">table</span>(portionanalyte)</a></code></pre></div>
<pre><code>portionanalyte
11R 12R 13R 21R 22R 31R 
393  35   1  26   2   2 </code></pre>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb62-1" data-line-number="1">samplevial &lt;-<span class="st"> </span><span class="kw">substr</span>(<span class="kw">colnames</span>(thca.filt4), <span class="dv">14</span>, <span class="dv">16</span>)</a>
<a class="sourceLine" id="cb62-2" data-line-number="2"><span class="kw">table</span>(samplevial)</a></code></pre></div>
<pre><code>samplevial
01A 01B 11A 11B 11C 
404   5  48   1   1 </code></pre>
<p>From this results, we can make the following observations:</p>
<ul>
<li><p>Samples were collected across different tissue source sites (TSS), altought there are several samples for each state it can be a potential surrogate variable as its distribution is not uniform.</p></li>
<li><p>All samples were sequenced at the same center, so it cannot have any confounding effect.</p></li>
<li><p>Different samples were sequenced with different plates, so it can also be a potential confunding variable, altough the sample distribution is almost uniform.</p></li>
<li><p>Most of the samples were sequenced using two portion and analyte combination, whereas the other combinations have few samples, so it also can be a surrogate variable.</p></li>
<li><p>There are six different combinations of tissue type and vial in the dataset, which can be a potential surrogate variable, as samples are not balanced distributed.</p></li>
</ul>
<p>From the TCGA barcode, we can extract four possible surrogate variables that can affect our outcome interesting variable (sample type). We summarized the information, recalculated the logCPM values with a higher prior cound in order to moderate extreme fold-changes produced by low counts, and performed a Hierchical Clustering plot in order to observe possible clustering due to the batch effect of the variables TSS (Figure <a href="quality-assessment.html#fig:sampleClustering">2.15</a>), plate (Figure <a href="quality-assessment.html#fig:sampleClustering2">2.16</a>), proportionanalyte (Figure <a href="quality-assessment.html#fig:sampleClustering3">2.17</a>) and samplevial (Figure <a href="quality-assessment.html#fig:sampleClustering4">2.18</a>).</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb64-1" data-line-number="1"><span class="kw">table</span>(<span class="kw">data.frame</span>(<span class="dt">TYPE=</span>thca.filt4<span class="op">$</span>type, <span class="dt">TSS=</span>tss))</a></code></pre></div>
<pre><code>        TSS
TYPE     4C BJ CE DE DJ DO E3 E8 EL EM ET FE FK FY GE H2 IM J8 KS L6 MK
  normal  0  8  0  0  0  1  0  1 27  5  3  0  0  1  0  2  0  0  2  0  0
  tumor   1 38  9  9 80  3  7 14 70 62 43 12  6 22  1  6  6  9  5  3  3</code></pre>
<div class="figure" style="text-align: center"><span id="fig:sampleClustering"></span>
<img src="QAanalysis_files/figure-html/sampleClustering-1.png" alt="Hierarchical clustering of the samples by TSS." width="800px" />
<p class="caption">
Figure 2.15: Hierarchical clustering of the samples by TSS.
</p>
</div>
<p>From the table, we can observed that normal tissues are usually under-represented with respect to the tumor tissues, that may lead to a potential confounding effect. However, the figure shows that there is not evident clustering with that factor, so we discard it as a possible surrogate variable.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb66-1" data-line-number="1"><span class="kw">table</span>(<span class="kw">data.frame</span>(<span class="dt">TYPE=</span>thca.filt4<span class="op">$</span>type, <span class="dt">plate=</span>plate))</a></code></pre></div>
<pre><code>        plate
TYPE     A10U A13Y A14Y A16R A180 A18C A19O A206 A20F A21D A220 A22L A22U
  normal    0    5    0    1    3    4    2    0    5    4    4    8    2
  tumor    20   11   17   19   16   18   53   34   21   20   21   34   12
        plate
TYPE     A23N A23W A250 A39I
  normal   12    0    0    0
  tumor    49   18   43    3</code></pre>
<div class="figure" style="text-align: center"><span id="fig:sampleClustering2"></span>
<img src="QAanalysis_files/figure-html/sampleClustering2-1.png" alt="Hierarchical clustering of the samples by Plate." width="800px" />
<p class="caption">
Figure 2.16: Hierarchical clustering of the samples by Plate.
</p>
</div>
<p>Altough the table shows different distribution across plates, it is mainly due to the difference size between tumor and normal samples. The plot shows that there is not clustering, so we also discard this factor as a surrogate variable.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb68-1" data-line-number="1"><span class="kw">table</span>(<span class="kw">data.frame</span>(<span class="dt">TYPE=</span>thca.filt4<span class="op">$</span>type, <span class="dt">PA=</span>portionanalyte))</a></code></pre></div>
<pre><code>        PA
TYPE     11R 12R 13R 21R 22R 31R
  normal  40   7   1   2   0   0
  tumor  353  28   0  24   2   2</code></pre>
<div class="figure" style="text-align: center"><span id="fig:sampleClustering3"></span>
<img src="QAanalysis_files/figure-html/sampleClustering3-1.png" alt="Hierarchical clustering of the samples by Portion Analytic." width="800px" />
<p class="caption">
Figure 2.17: Hierarchical clustering of the samples by Portion Analytic.
</p>
</div>
<p>This plot also does not show a significant clustering, so we also discard this variable.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb70-1" data-line-number="1"><span class="kw">table</span>(<span class="kw">data.frame</span>(<span class="dt">TYPE=</span>thca.filt4<span class="op">$</span>type, <span class="dt">SV=</span>samplevial))</a></code></pre></div>
<pre><code>        SV
TYPE     01A 01B 11A 11B 11C
  normal   0   0  48   1   1
  tumor  404   5   0   0   0</code></pre>
<div class="figure" style="text-align: center"><span id="fig:sampleClustering4"></span>
<img src="QAanalysis_files/figure-html/sampleClustering4-1.png" alt="Hierarchical clustering of the samples by Sample vial." width="800px" />
<p class="caption">
Figure 2.18: Hierarchical clustering of the samples by Sample vial.
</p>
</div>
<p>From this plot, we can observe that there is a clustering of the samples according to samplevial, which can interfer with our outcome of interest. Moreover, we perform a MDS plot (Figure <a href="quality-assessment.html#fig:mdsPlot">2.19</a>) to visualize the dependency.</p>
<div class="figure" style="text-align: center"><span id="fig:mdsPlot"></span>
<img src="QAanalysis_files/figure-html/mdsPlot-1.png" alt="Multidimensional scaling plot of the samples. Shows clustering of samples by fold change in logarithmic scale, coloured by sample vial." width="800px" />
<p class="caption">
Figure 2.19: Multidimensional scaling plot of the samples. Shows clustering of samples by fold change in logarithmic scale, coloured by sample vial.
</p>
</div>
<p>Samplevial seems to cause a confunding effect with the outcome of interest, so it is considered as a real surrogate variable. We would need to adjust or remove this batch effect in order to get a non-biased and trustworthy result.</p>
</div>
<div id="session-information" class="section level2">
<h2><span class="header-section-number">2.9</span> Session information</h2>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb72-1" data-line-number="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>R version 3.5.0 (2018-04-23)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.4

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] es_ES.UTF-8/es_ES.UTF-8/es_ES.UTF-8/C/es_ES.UTF-8/es_ES.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] geneplotter_1.58.0          annotate_1.58.0            
 [3] XML_3.98-1.11               AnnotationDbi_1.42.1       
 [5] lattice_0.20-35             edgeR_3.22.1               
 [7] limma_3.36.1                SummarizedExperiment_1.10.0
 [9] DelayedArray_0.6.0          BiocParallel_1.14.1        
[11] matrixStats_0.53.1          Biobase_2.40.0             
[13] GenomicRanges_1.32.2        GenomeInfoDb_1.16.0        
[15] IRanges_2.14.5              S4Vectors_0.18.1           
[17] BiocGenerics_0.26.0         BiocStyle_2.8.0            
[19] knitr_1.20                 

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.16           highr_0.6              RColorBrewer_1.1-2    
 [4] compiler_3.5.0         XVector_0.20.0         bitops_1.0-6          
 [7] tools_3.5.0            zlibbioc_1.26.0        bit_1.1-12            
[10] digest_0.6.15          memoise_1.1.0          evaluate_0.10.1       
[13] RSQLite_2.1.1          Matrix_1.2-14          DBI_1.0.0             
[16] yaml_2.1.19            xfun_0.1               GenomeInfoDbData_1.1.0
[19] stringr_1.3.0          bit64_0.9-7            locfit_1.5-9.1        
[22] rprojroot_1.3-2        grid_3.5.0             rmarkdown_1.9         
[25] bookdown_0.7           blob_1.1.1             magrittr_1.5          
[28] codetools_0.2-15       backports_1.1.2        htmltools_0.3.6       
[31] xtable_1.8-2           KernSmooth_2.23-15     stringi_1.2.2         
[34] RCurl_1.95-4.10       </code></pre>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="visualising-data.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
},
"search": false
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
