<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>QAanalysis.utf8</title>
  <meta name="description" content="">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="QAanalysis.utf8" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="QAanalysis.utf8" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  

<link rel="next" href="quality-assessment.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="visualising-data.html"><a href="visualising-data.html"><i class="fa fa-check"></i><b>1</b> Visualising data</a></li>
<li class="chapter" data-level="2" data-path="quality-assessment.html"><a href="quality-assessment.html"><i class="fa fa-check"></i><b>2</b> Quality assessment</a><ul>
<li class="chapter" data-level="2.1" data-path="quality-assessment.html"><a href="quality-assessment.html#important-variables"><i class="fa fa-check"></i><b>2.1</b> Important Variables</a></li>
<li class="chapter" data-level="2.2" data-path="quality-assessment.html"><a href="quality-assessment.html#sequencing-depth"><i class="fa fa-check"></i><b>2.2</b> Sequencing depth</a></li>
<li class="chapter" data-level="2.3" data-path="quality-assessment.html"><a href="quality-assessment.html#distribution-of-expression-levels-among-samples"><i class="fa fa-check"></i><b>2.3</b> Distribution of expression levels among samples</a></li>
<li class="chapter" data-level="2.4" data-path="quality-assessment.html"><a href="quality-assessment.html#distribution-of-expression-levels-among-genes"><i class="fa fa-check"></i><b>2.4</b> Distribution of expression levels among genes</a></li>
<li class="chapter" data-level="2.5" data-path="quality-assessment.html"><a href="quality-assessment.html#filtering-of-lowly-expressed-genes"><i class="fa fa-check"></i><b>2.5</b> Filtering of lowly-expressed genes</a></li>
<li class="chapter" data-level="2.6" data-path="quality-assessment.html"><a href="quality-assessment.html#normalization"><i class="fa fa-check"></i><b>2.6</b> Normalization</a></li>
<li class="chapter" data-level="2.7" data-path="quality-assessment.html"><a href="quality-assessment.html#ma-plots"><i class="fa fa-check"></i><b>2.7</b> MA-plots</a></li>
<li class="chapter" data-level="2.8" data-path="quality-assessment.html"><a href="quality-assessment.html#batch-identification"><i class="fa fa-check"></i><b>2.8</b> Batch identification</a></li>
<li class="chapter" data-level="2.9" data-path="quality-assessment.html"><a href="quality-assessment.html#session-information"><i class="fa fa-check"></i><b>2.9</b> Session information</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<!--bookdown:title:end-->
<!--bookdown:title:start-->
<div id="visualising-data" class="section level1">
<h1><span class="header-section-number">1</span> Visualising data</h1>
<p>The data under study consists in counts of reads for the Thyroid carcinoma as a Summarized Experiment object. This dataset comes from the “Alternative preprocessing of RNA-Sequencing data in The Cancer Genome Atlas leads to improved analysis results” project, with 20115 analyzed genes and 572 samples.</p>
<p>The column data contains pehnotypic data, which corresponds to clinical variables and their corresponding metadata, it contains 549 columns with different information about the samples.</p>
<p>The metadata consists of two columns of information about the clinical variables. One called <code>labelDescription</code> contains a succint description of the variable, often
not more self-explanatory than the variable name itself, and the other called ‘CDEID’ corresponds to the so-called <code>Common Data Element (CDE)</code> identifier. This
identifier can be use in <a href="https://cdebrowser.nci.nih.gov" class="uri">https://cdebrowser.nci.nih.gov</a> to search for further information about the associated clinical variable using the <code>Advanced search</code>
form and the <code>Public ID</code> attribute search.</p>
<p>The row data contains feature data, with the characteristics of the studied genes such as their location, ranges, strand, symbol, length and CG content.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">thca &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">file.path</span>(<span class="st">&quot;raw_counts&quot;</span>, <span class="st">&quot;seTHCA.rds&quot;</span>))</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">thca</a></code></pre></div>
<pre><code>class: RangedSummarizedExperiment 
dim: 20115 572 
metadata(5): experimentData annotation cancerTypeCode
  cancerTypeDescription objectCreationDate
assays(1): counts
rownames(20115): 1 2 ... 102724473 103091865
rowData names(3): symbol txlen txgc
colnames(572): TCGA.4C.A93U.01A.11R.A39I.07
  TCGA.BJ.A0YZ.01A.11R.A10U.07 ... TCGA.KS.A41J.11A.12R.A23N.07
  TCGA.KS.A41L.11A.11R.A23N.07
colData names(549): type bcr_patient_uuid ...
  lymph_nodes_aortic_pos_by_ihc lymph_nodes_aortic_pos_total</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw">dim</span>(<span class="kw">colData</span>(thca))</a></code></pre></div>
<pre><code>[1] 572 549</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">colData</span>(thca)[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</a></code></pre></div>
<pre><code>DataFrame with 5 rows and 5 columns
                                 type                     bcr_patient_uuid
                             &lt;factor&gt;                             &lt;factor&gt;
TCGA.4C.A93U.01A.11R.A39I.07    tumor E1C8AF06-9EA4-4062-9264-AA916E0025D1
TCGA.BJ.A0YZ.01A.11R.A10U.07    tumor f3ee888e-5116-4313-a2f6-3b1dcc3e4bc1
TCGA.BJ.A0Z0.01A.11R.A10U.07    tumor 72d8dcc3-0709-4fd1-98d4-fb75e9340758
TCGA.BJ.A0Z2.01A.11R.A10U.07    tumor f9ceffc0-d544-418d-b4a9-bd3c84e37026
TCGA.BJ.A0Z3.01A.11R.A13Y.07    tumor 331cae6e-2868-4c58-9302-709a9ff7d025
                             bcr_patient_barcode form_completion_date
                                        &lt;factor&gt;             &lt;factor&gt;
TCGA.4C.A93U.01A.11R.A39I.07        TCGA-4C-A93U           2014-11-12
TCGA.BJ.A0YZ.01A.11R.A10U.07        TCGA-BJ-A0YZ            2011-4-11
TCGA.BJ.A0Z0.01A.11R.A10U.07        TCGA-BJ-A0Z0            2011-4-15
TCGA.BJ.A0Z2.01A.11R.A10U.07        TCGA-BJ-A0Z2            2011-4-15
TCGA.BJ.A0Z3.01A.11R.A13Y.07        TCGA-BJ-A0Z3             2011-6-2
                             prospective_collection
                                           &lt;factor&gt;
TCGA.4C.A93U.01A.11R.A39I.07                    YES
TCGA.BJ.A0YZ.01A.11R.A10U.07                     NO
TCGA.BJ.A0Z0.01A.11R.A10U.07                     NO
TCGA.BJ.A0Z2.01A.11R.A10U.07                     NO
TCGA.BJ.A0Z3.01A.11R.A13Y.07                     NO</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">mcols</span>(<span class="kw">colData</span>(thca), <span class="dt">use.names=</span><span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>DataFrame with 549 rows and 2 columns
                                                         labelDescription
                                                              &lt;character&gt;
type                                           sample type (tumor/normal)
bcr_patient_uuid                                         bcr patient uuid
bcr_patient_barcode                                   bcr patient barcode
form_completion_date                                 form completion date
prospective_collection            tissue prospective collection indicator
...                                                                   ...
lymph_nodes_pelvic_pos_total                               total pelv lnp
lymph_nodes_aortic_examined_count                           total aor lnr
lymph_nodes_aortic_pos_by_he                          aln pos light micro
lymph_nodes_aortic_pos_by_ihc                                 aln pos ihc
lymph_nodes_aortic_pos_total                                total aor-lnp
                                        CDEID
                                  &lt;character&gt;
type                                       NA
bcr_patient_uuid                           NA
bcr_patient_barcode                   2673794
form_completion_date                       NA
prospective_collection                3088492
...                                       ...
lymph_nodes_pelvic_pos_total          3151828
lymph_nodes_aortic_examined_count     3104460
lymph_nodes_aortic_pos_by_he          3151832
lymph_nodes_aortic_pos_by_ihc         3151831
lymph_nodes_aortic_pos_total          3151827</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">metadata</span>(thca)</a></code></pre></div>
<pre><code>$experimentData
Experiment data
  Experimenter name: Piccolo SR 
  Laboratory: Dept. of Biology, Brigham Young University, Provo, UT (USA) 
  Contact information: stephen_piccolo@byu.edu 
  Title: Alternative preprocessing of RNA-Sequencing data in The Cancer Genome Atlas leads to improved analysis results 
  URL: https://github.com/srp33/TCGA_RNASeq_Clinical 
  PMIDs: 26209429 
  No abstract available.

$annotation
[1] &quot;org.Hs.eg.db&quot;

$cancerTypeCode
[1] &quot;THCA&quot;

$cancerTypeDescription
[1] &quot;Thyroid carcinoma&quot;

$objectCreationDate
[1] &quot;2016-04-25&quot;</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw">rowData</span>(thca)</a></code></pre></div>
<pre><code>DataFrame with 20115 rows and 3 columns
           symbol     txlen              txgc
      &lt;character&gt; &lt;integer&gt;         &lt;numeric&gt;
1            A1BG      3322 0.564419024683925
2             A2M      4844 0.488232865400495
3            NAT1      2280 0.394298245614035
4            NAT2      1322 0.389561270801815
5        SERPINA3      3067 0.524942940984676
...           ...       ...               ...
20111       POTEB      1706 0.430832356389215
20112    SNORD124       104 0.490384615384615
20113   SNORD121B        81 0.407407407407407
20114      GAGE10       538 0.505576208178439
20115   BRWD1-IT2      1028 0.592412451361868</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw">rowRanges</span>(thca)</a></code></pre></div>
<pre><code>GRanges object with 20115 ranges and 3 metadata columns:
            seqnames            ranges strand |      symbol     txlen
               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;integer&gt;
          1    chr19 58345178-58362751      - |        A1BG      3322
          2    chr12   9067664-9116229      - |         A2M      4844
          9     chr8 18170477-18223689      + |        NAT1      2280
         10     chr8 18391245-18401218      + |        NAT2      1322
         12    chr14 94592058-94624646      + |    SERPINA3      3067
        ...      ...               ...    ... .         ...       ...
  100996331    chr15 20835372-21877298      - |       POTEB      1706
  101340251    chr17 40027542-40027645      - |    SNORD124       104
  101340252     chr9 33934296-33934376      - |   SNORD121B        81
  102724473     chrX 49303669-49319844      + |      GAGE10       538
  103091865    chr21 39313935-39314962      + |   BRWD1-IT2      1028
                         txgc
                    &lt;numeric&gt;
          1 0.564419024683925
          2 0.488232865400495
          9 0.394298245614035
         10 0.389561270801815
         12 0.524942940984676
        ...               ...
  100996331 0.430832356389215
  101340251 0.490384615384615
  101340252 0.407407407407407
  102724473 0.505576208178439
  103091865 0.592412451361868
  -------
  seqinfo: 455 sequences (1 circular) from hg38 genome</code></pre>
<p>In this supplementary material we will perform a quality assessment and normalization of the data. To do that, we first will create a ‘DGEList’ object, and the <span class="math inline">\(\log_2\)</span> CPM values of expression are calculated in order to ease the manipulation of the data.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">dge &lt;-<span class="st"> </span><span class="kw">DGEList</span>(<span class="dt">counts=</span><span class="kw">assays</span>(thca)<span class="op">$</span>counts, <span class="dt">genes=</span><span class="kw">mcols</span>(thca))</a>
<a class="sourceLine" id="cb15-2" data-line-number="2"><span class="kw">saveRDS</span>(dge, <span class="kw">file.path</span>(<span class="st">&quot;results&quot;</span>, <span class="st">&quot;dge.rds&quot;</span>))</a></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw">assays</span>(thca)<span class="op">$</span>logCPM &lt;-<span class="st"> </span><span class="kw">cpm</span>(dge, <span class="dt">log=</span><span class="ot">TRUE</span>, <span class="dt">prior.count=</span><span class="fl">0.5</span>)</a>
<a class="sourceLine" id="cb16-2" data-line-number="2"><span class="kw">assays</span>(thca)<span class="op">$</span>logCPM[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</a></code></pre></div>
<pre><code>   TCGA.4C.A93U.01A.11R.A39I.07 TCGA.BJ.A0YZ.01A.11R.A10U.07
1                      2.408053                     3.875800
2                      9.483176                    10.203685
9                     -6.667872                    -6.667872
10                    -6.667872                    -6.667872
12                     3.688878                     4.988388
   TCGA.BJ.A0Z0.01A.11R.A10U.07 TCGA.BJ.A0Z2.01A.11R.A10U.07
1                      2.778138                     4.239719
2                      8.890882                     8.403990
9                     -6.667872                    -6.667872
10                    -6.667872                    -6.667872
12                     4.804717                     4.839157
   TCGA.BJ.A0Z3.01A.11R.A13Y.07
1                      2.655578
2                      9.190160
9                     -6.667872
10                    -6.667872
12                     4.605054</code></pre>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="quality-assessment.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
},
"search": false
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
